
<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
    <meta http-equiv="x-ua-compatible" content="ie=edge">
    <meta name="author" content="Nejc Zupan">
    <link rel="icon" type="image/png" href="/static/img/favicon.png">
    <title>Kipoi</title>
    <!-- Font Awesome -->
<link rel="stylesheet" type="text/css" href="/static/css/font-awesome.min.css">
<!-- Bootstrap core CSS -->
<link rel="stylesheet" type="text/css" href="/static/css/bootstrap.min.css">
<!-- Material Design Bootstrap -->
<link rel="stylesheet" type="text/css" href="/static/css/mdb.min.css">
<!-- Material Design Lite -->
<link rel="stylesheet" type="text/css" href="/static/css/material.min.css">
<!-- DataTables Material Design -->    
<link rel="stylesheet" type="text/css" href="/static/css/dataTables.material.css"/>
<!-- Highlight -->
<link rel="stylesheet" type="text/css" href="/static/css/custom_highlight_github.css"/>

<!-- Custom styles -->
<link rel="stylesheet" type="text/css" href="/static/css/style.css?v=150318">
</head>

<body>
    <section class="mb-3">
        
<!--Navbar-->

<nav class="navbar fixed-top navbar-expand-sm navbar-dark unique-color-kipoi">

    <!-- Navbar brand -->
    <span>
      <a class="nav-title-link navbar-brand-nopad text-hide left-margin" href="/"
	 ><img src="/static/img/nav_bar_logo_v3.png" height="50px" alt="Kipoi"/></a>
    </span>

    <!-- Collapse button -->
    <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarNav1" aria-controls="navbarNav1" aria-expanded="false" aria-label="Toggle navigation"><span class="navbar-toggler-icon"></span></button>
    
    <!-- Collapsible content -->
    <div class="collapse navbar-collapse text-center" id="navbarNav1">
        
        <!-- Links -->
        <ul class="navbar-nav mr-auto text-center">
	  <!-- Group list --> 
            <li class="nav-item">
              <a id="components-related-1" class="nav-button btn btn-outline-primary navbar-dark nabar-button " href="/groups/">Models</a>
            </li>
            <!-- About --> 
            <li class="nav-item">
              <a id="components-related-2" class="nav-button btn btn-outline-primary navbar-dark nabar-button " href="/about/">About</a>
            </li>
	    <!-- Docs --> 
            <li class="nav-item">
              <a id="components-related-3" class="nav-button btn btn-outline-primary navbar-dark nabar-button" href="/docs/">Docs</a>
            </li>
        <!-- Seminar -->
            <li class="nav-item">
              <a id="components-related-4" class="nav-button btn btn-outline-primary navbar-dark nabar-button " href="/seminar/">Seminar</a>
            </li>
        </ul>

        <!-- Search form -->
        

      
        <ul class="navbar-nav ml-auto text-center d-sm-none d-md-flex">
      
          <li class="nav-item">
            <a id="components-related-5" class="text-nowrap nav-link waves-effect waves-light" href="https://www.biorxiv.org/content/early/2018/07/24/375345.full.pdf">White paper<i class="fa fa-file-text-o" aria-hidden="true"></i></a>
          </li>
          <li class="nav-item">
            <a id="components-related-6" class="text-nowrap nav-link waves-effect waves-light" href="https://github.com/kipoi/models">Model repository<i class="fa fa-github" aria-hidden="true"></i></a>
          </li>
          <li class="nav-item">
            <a id="components-related-7" class="text-nowrap nav-link waves-effect waves-light" href="https://github.com/kipoi/kipoi">API repository<i class="fa fa-github" aria-hidden="true"></i></a>
          </li>
        </ul>
        
    </div>
</nav>
    </section>

    <section class="main-content">
        
    <div class="col-lg-12 col-md-12 col-sm-12">
        <!--Section: model-details content-->
        <section class="model-details">
            <!--<h2 class="">Model Details</h2>-->

                <div class="jumbotron">
                    <div class="row">
                        <div class="col-lg-7">
                            <!-- Model title -->
                            <h1 class="h1-responsive"><a href="/models/DeepBind">DeepBind</a>/<a href="/models/DeepBind/Mus_musculus">Mus_musculus</a>/<a href="/models/DeepBind/Mus_musculus/TF">TF</a>/D00076.001_PBM_Sp140</h1>
                        </div>
                        <!-- Model details -->
                        <div class="col-lg-5">
                            <div class="row">
                                <div class="col-lg-6 col-md-6 col-sm-6 col-xs-12">
                                    <p>Authors: 
  
    
      
  Babak Alipanahi
  
  

    
    , 
  
    
      
  Andrew Delong
  
  

    
    , 
  
    
      
  Matthew T Weirauch
  
  

    
    , 
  
    
      
  Brendan J Frey
  
  

    
    
  
</p>
                                    <!-- <p>
                                         Name:
                                         
                                         DeepBind
                                         
					 </p> -->
                                    <!-- <p>Version: 0.1</p> -->
                                    <!-- License -->
                                    <p>License: BSD 3-Clause</p>
                                    <p>Contributed by: 
  
    
      
  Johnny Israeli
  
    <a href="https://github.com/jisraeli" class="github">
      <i class="fa fa-github" aria-hidden="true"></i>
    </a>
  
  

    
    
  
</p>
                                    <!-- Cite as -->
                                    <p>Cite as: 
  
  
      
  
    <a href="https://doi.org/10.1038/nbt.3300">https://doi.org/10.1038/nbt.3300</a>
  

    
  
</p>
                                </div>
                                <div class="col-lg-6 col-md-6 col-xs-12">
                                    <!-- Model type -->
                                    <p>Type:
                                        
                                      <a data-toggle="tooltip" data-placement="top" data-html="true" title="
  <div>
    
      <strong>arch</strong>: RemoteFile(url=&#39;https://sandbox.zenodo.org/record/248887/files/Mus_musculus-TF-D00076.001_PBM_Sp140?download=1&#39;, md5=&#39;bf5fc00b641e9958ed3d652b9e8a343e&#39;, name=&#39;&#39;)<br>
    
      <strong>weights</strong>: RemoteFile(url=&#39;https://sandbox.zenodo.org/record/248887/files/Mus_musculus-TF-D00076.001_PBM_Sp140.h5?download=1&#39;, md5=&#39;f03a799973742794037f83b11b00228f&#39;, name=&#39;&#39;)<br>
    
      <strong>custom_objects</strong>: custom_keras_objects.py<br>
    
      <strong>backend</strong>: tensorflow<br>
    
      <strong>image_dim_ordering</strong>: tf<br>
    
  </div>
"><span class="badge badge-pill rgba-stylish-strong hoverable" >keras</span></a>
                                      
                                    </p>

                                    <!-- Model postprocessing -->
                                    <p>
                                      Postprocessing:
                                      
                                          <a data-toggle="tooltip" data-placement="top" data-html="true"><span class="badge badge-pill rgba-stylish-strong hoverable" >variant_effects</span></a>
                                      
                                    </p>
				    
                                    <!-- Trained on -->
                                    <p>Trained on: 
  
    ?All chromosomes? Data from protein binding microarrays (Mukherjee et al., 2004), RNAcompete assays (Ray et al., 2009), ChIP-seq (Kharchenko et al., 2008), and HT-SELEX (Jolma et al., 2010)
  
</p>

				    <!-- Model url  -->
				    <p>
				      <a href="https://github.com/kipoi/models/tree/master/DeepBind">Source files <i class="fa fa-github" aria-hidden="true"></i></a>
				    </p> 
                                </div>
                            </div>
                        </div>
                    </div>
                    <div class="row">
                        <div class="col-lg-7 col-md-12 col-sm-12">
                            <!-- Model description -->
                            <p class="lead">Abstract: Knowing the sequence specificities of DNA- and RNA-binding proteins is essential for developing models of the regulatory processes in biological systems and for identifying causal disease variants. Here we show that sequence specificities can be ascertained from experimental data with &#39;deep learning&#39; techniques, which offer a scalable, flexible and unified computational approach for pattern discovery. Using a diverse array of experimental data and evaluation metrics, we find that deep learning outperforms other state-of-the-art methods, even when training on in vitro data and testing on in vivo data. We call this approach DeepBind and have built a stand-alone software tool that is fully automatic and handles millions of sequences per experiment. Specificities determined by DeepBind are readily visualized as a weighted ensemble of position weight matrices or as a &#39;mutation map&#39; that indicates how variations affect binding within a specific sequence.
</p>
							<p></p>
                        </div>

                        <div class="col-lg-5 col-md-12 col-sm-12">
                            <!-- Model code snippets -->
                            <ul class="nav nav-tabs" role="tablist">
    <li class="nav-item">
        <a class="nav-link active" data-toggle="tab" href="#cli" role="tab">CLI</a>
    </li>
    <li class="nav-item">
        <a class="nav-link" data-toggle="tab" href="#python" role="tab">python</a>
    </li>
    <li class="nav-item">
        <a class="nav-link" data-toggle="tab" href="#r" role="tab">R</a>
    </li>
    <li class="nav-item">
        <a class="nav-link" data-toggle="tab" href="#docker" role="tab">docker</a>
    </li>
    <li class="nav-item">
        <a class="nav-link" data-toggle="tab" href="#singularity" role="tab">singularity</a>
    </li>
</ul>

<!-- Tab panes -->
<div class="tab-content">
    <!-- CLI code snippet tab -->
    <div class="tab-pane active" id="cli" role="tabpanel">
        <div class="border border-top-0 code-snippet-container">
            
                
  <div class="row">
    <div class="col-lg-12 col-xs-12">
      <div class="code-title">Create a new conda environment with all dependencies installed</div>
    </div>
    <div class="col-lg-12 col-xs-12">
      <div class="code-container">
	<!-- <pre class="language-markup"><code>kipoi env create DeepBind
source activate kipoi-DeepBind</code></pre> -->
	<pre><code class="bash">kipoi env create DeepBind
source activate kipoi-DeepBind</code></pre>
	<div class="code-toolbar"><a data-toggle="tooltip" data-placement="top" data-html="true" title="Copy to clipboard!" class="btn btn-sm btn-info copy-to-clipboard">Copy</a></div>
      </div>
    </div>
  </div>

            
                
  <div class="row">
    <div class="col-lg-12 col-xs-12">
      <div class="code-title">Test the model</div>
    </div>
    <div class="col-lg-12 col-xs-12">
      <div class="code-container">
	<!-- <pre class="language-markup"><code>kipoi test DeepBind/Mus_musculus/TF/D00076.001_PBM_Sp140 --source=kipoi</code></pre> -->
	<pre><code class="bash">kipoi test DeepBind/Mus_musculus/TF/D00076.001_PBM_Sp140 --source=kipoi</code></pre>
	<div class="code-toolbar"><a data-toggle="tooltip" data-placement="top" data-html="true" title="Copy to clipboard!" class="btn btn-sm btn-info copy-to-clipboard">Copy</a></div>
      </div>
    </div>
  </div>

            
                
  <div class="row">
    <div class="col-lg-12 col-xs-12">
      <div class="code-title">Make a prediction</div>
    </div>
    <div class="col-lg-12 col-xs-12">
      <div class="code-container">
	<!-- <pre class="language-markup"><code>kipoi get-example DeepBind/Mus_musculus/TF/D00076.001_PBM_Sp140 -o example
kipoi predict DeepBind/Mus_musculus/TF/D00076.001_PBM_Sp140 \
  --dataloader_args=&#39;{&#34;intervals_file&#34;: &#34;example/intervals_file&#34;, &#34;fasta_file&#34;: &#34;example/fasta_file&#34;}&#39; \
  -o &#39;/tmp/DeepBind|Mus_musculus|TF|D00076.001_PBM_Sp140.example_pred.tsv&#39;
# check the results
head &#39;/tmp/DeepBind|Mus_musculus|TF|D00076.001_PBM_Sp140.example_pred.tsv&#39;
</code></pre> -->
	<pre><code class="bash">kipoi get-example DeepBind/Mus_musculus/TF/D00076.001_PBM_Sp140 -o example
kipoi predict DeepBind/Mus_musculus/TF/D00076.001_PBM_Sp140 \
  --dataloader_args=&#39;{&#34;intervals_file&#34;: &#34;example/intervals_file&#34;, &#34;fasta_file&#34;: &#34;example/fasta_file&#34;}&#39; \
  -o &#39;/tmp/DeepBind|Mus_musculus|TF|D00076.001_PBM_Sp140.example_pred.tsv&#39;
# check the results
head &#39;/tmp/DeepBind|Mus_musculus|TF|D00076.001_PBM_Sp140.example_pred.tsv&#39;
</code></pre>
	<div class="code-toolbar"><a data-toggle="tooltip" data-placement="top" data-html="true" title="Copy to clipboard!" class="btn btn-sm btn-info copy-to-clipboard">Copy</a></div>
      </div>
    </div>
  </div>

            
        </div>
    </div>

    <!-- python code snippet tab -->
    <div class="tab-pane" id="python" role="tabpanel">
        <div class="border border-top-0 code-snippet-container">
            
                
  <div class="row">
    <div class="col-lg-12 col-xs-12">
      <div class="code-title">Create a new conda environment with all dependencies installed</div>
    </div>
    <div class="col-lg-12 col-xs-12">
      <div class="code-container">
	<!-- <pre class="language-markup"><code>kipoi env create DeepBind
source activate kipoi-DeepBind</code></pre> -->
	<pre><code class="python">kipoi env create DeepBind
source activate kipoi-DeepBind</code></pre>
	<div class="code-toolbar"><a data-toggle="tooltip" data-placement="top" data-html="true" title="Copy to clipboard!" class="btn btn-sm btn-info copy-to-clipboard">Copy</a></div>
      </div>
    </div>
  </div>

            
                
  <div class="row">
    <div class="col-lg-12 col-xs-12">
      <div class="code-title">Get the model</div>
    </div>
    <div class="col-lg-12 col-xs-12">
      <div class="code-container">
	<!-- <pre class="language-markup"><code>import kipoi
model = kipoi.get_model(&#39;DeepBind/Mus_musculus/TF/D00076.001_PBM_Sp140&#39;)</code></pre> -->
	<pre><code class="python">import kipoi
model = kipoi.get_model(&#39;DeepBind/Mus_musculus/TF/D00076.001_PBM_Sp140&#39;)</code></pre>
	<div class="code-toolbar"><a data-toggle="tooltip" data-placement="top" data-html="true" title="Copy to clipboard!" class="btn btn-sm btn-info copy-to-clipboard">Copy</a></div>
      </div>
    </div>
  </div>

            
                
  <div class="row">
    <div class="col-lg-12 col-xs-12">
      <div class="code-title">Make a prediction for example files</div>
    </div>
    <div class="col-lg-12 col-xs-12">
      <div class="code-container">
	<!-- <pre class="language-markup"><code>pred = model.pipeline.predict_example(batch_size=4)</code></pre> -->
	<pre><code class="python">pred = model.pipeline.predict_example(batch_size=4)</code></pre>
	<div class="code-toolbar"><a data-toggle="tooltip" data-placement="top" data-html="true" title="Copy to clipboard!" class="btn btn-sm btn-info copy-to-clipboard">Copy</a></div>
      </div>
    </div>
  </div>

            
                
  <div class="row">
    <div class="col-lg-12 col-xs-12">
      <div class="code-title">Use dataloader and model separately</div>
    </div>
    <div class="col-lg-12 col-xs-12">
      <div class="code-container">
	<!-- <pre class="language-markup"><code># Download example dataloader kwargs
dl_kwargs = model.default_dataloader.download_example(&#39;example&#39;)
# Get the dataloader and instantiate it
dl = model.default_dataloader(**dl_kwargs)
# get a batch iterator
batch_iterator = dl.batch_iter(batch_size=4)
for batch in batch_iterator:
    # predict for a batch
    batch_pred = model.predict_on_batch(batch[&#39;inputs&#39;])</code></pre> -->
	<pre><code class="python"># Download example dataloader kwargs
dl_kwargs = model.default_dataloader.download_example(&#39;example&#39;)
# Get the dataloader and instantiate it
dl = model.default_dataloader(**dl_kwargs)
# get a batch iterator
batch_iterator = dl.batch_iter(batch_size=4)
for batch in batch_iterator:
    # predict for a batch
    batch_pred = model.predict_on_batch(batch[&#39;inputs&#39;])</code></pre>
	<div class="code-toolbar"><a data-toggle="tooltip" data-placement="top" data-html="true" title="Copy to clipboard!" class="btn btn-sm btn-info copy-to-clipboard">Copy</a></div>
      </div>
    </div>
  </div>

            
                
  <div class="row">
    <div class="col-lg-12 col-xs-12">
      <div class="code-title">Make predictions for custom files directly</div>
    </div>
    <div class="col-lg-12 col-xs-12">
      <div class="code-container">
	<!-- <pre class="language-markup"><code>pred = model.pipeline.predict(dl_kwargs, batch_size=4)</code></pre> -->
	<pre><code class="python">pred = model.pipeline.predict(dl_kwargs, batch_size=4)</code></pre>
	<div class="code-toolbar"><a data-toggle="tooltip" data-placement="top" data-html="true" title="Copy to clipboard!" class="btn btn-sm btn-info copy-to-clipboard">Copy</a></div>
      </div>
    </div>
  </div>

            
        </div>
    </div>
    
    <!-- R code snippet tab -->
    <div class="tab-pane" id="r" role="tabpanel">
        <div class="border border-top-0 code-snippet-container">
            
                
  <div class="row">
    <div class="col-lg-12 col-xs-12">
      <div class="code-title">Get the model</div>
    </div>
    <div class="col-lg-12 col-xs-12">
      <div class="code-container">
	<!-- <pre class="language-markup"><code>library(reticulate)
kipoi &lt;- import(&#39;kipoi&#39;)
model &lt;- kipoi$get_model(&#39;DeepBind/Mus_musculus/TF/D00076.001_PBM_Sp140&#39;)</code></pre> -->
	<pre><code class="R">library(reticulate)
kipoi &lt;- import(&#39;kipoi&#39;)
model &lt;- kipoi$get_model(&#39;DeepBind/Mus_musculus/TF/D00076.001_PBM_Sp140&#39;)</code></pre>
	<div class="code-toolbar"><a data-toggle="tooltip" data-placement="top" data-html="true" title="Copy to clipboard!" class="btn btn-sm btn-info copy-to-clipboard">Copy</a></div>
      </div>
    </div>
  </div>

            
                
  <div class="row">
    <div class="col-lg-12 col-xs-12">
      <div class="code-title">Make a prediction for example files</div>
    </div>
    <div class="col-lg-12 col-xs-12">
      <div class="code-container">
	<!-- <pre class="language-markup"><code>predictions &lt;- model$pipeline$predict_example()</code></pre> -->
	<pre><code class="R">predictions &lt;- model$pipeline$predict_example()</code></pre>
	<div class="code-toolbar"><a data-toggle="tooltip" data-placement="top" data-html="true" title="Copy to clipboard!" class="btn btn-sm btn-info copy-to-clipboard">Copy</a></div>
      </div>
    </div>
  </div>

            
                
  <div class="row">
    <div class="col-lg-12 col-xs-12">
      <div class="code-title">Use dataloader and model separately</div>
    </div>
    <div class="col-lg-12 col-xs-12">
      <div class="code-container">
	<!-- <pre class="language-markup"><code># Download example dataloader kwargs
dl_kwargs &lt;- model$default_dataloader$download_example(&#39;example&#39;)
# Get the dataloader
dl &lt;- model$default_dataloader(dl_kwargs)
# get a batch iterator
it &lt;- dl$batch_iter(batch_size=4)
# predict for a batch
batch &lt;- iter_next(it)
model$predict_on_batch(batch$inputs)</code></pre> -->
	<pre><code class="R"># Download example dataloader kwargs
dl_kwargs &lt;- model$default_dataloader$download_example(&#39;example&#39;)
# Get the dataloader
dl &lt;- model$default_dataloader(dl_kwargs)
# get a batch iterator
it &lt;- dl$batch_iter(batch_size=4)
# predict for a batch
batch &lt;- iter_next(it)
model$predict_on_batch(batch$inputs)</code></pre>
	<div class="code-toolbar"><a data-toggle="tooltip" data-placement="top" data-html="true" title="Copy to clipboard!" class="btn btn-sm btn-info copy-to-clipboard">Copy</a></div>
      </div>
    </div>
  </div>

            
                
  <div class="row">
    <div class="col-lg-12 col-xs-12">
      <div class="code-title">Make predictions for custom files directly</div>
    </div>
    <div class="col-lg-12 col-xs-12">
      <div class="code-container">
	<!-- <pre class="language-markup"><code>pred &lt;- model$pipeline$predict(dl_kwargs, batch_size=4)</code></pre> -->
	<pre><code class="R">pred &lt;- model$pipeline$predict(dl_kwargs, batch_size=4)</code></pre>
	<div class="code-toolbar"><a data-toggle="tooltip" data-placement="top" data-html="true" title="Copy to clipboard!" class="btn btn-sm btn-info copy-to-clipboard">Copy</a></div>
      </div>
    </div>
  </div>

            
        </div>
    </div>

    <!-- docker code snippet tab -->
    <div class="tab-pane" id="docker" role="tabpanel">
        <div class="border border-top-0 code-snippet-container">
            
                
  <div class="row">
    <div class="col-lg-12 col-xs-12">
      <div class="code-title">Get the docker image</div>
    </div>
    <div class="col-lg-12 col-xs-12">
      <div class="code-container">
	<!-- <pre class="language-markup"><code>docker pull kipoi/kipoi-docker:sharedpy3keras2</code></pre> -->
	<pre><code class="docker">docker pull kipoi/kipoi-docker:sharedpy3keras2</code></pre>
	<div class="code-toolbar"><a data-toggle="tooltip" data-placement="top" data-html="true" title="Copy to clipboard!" class="btn btn-sm btn-info copy-to-clipboard">Copy</a></div>
      </div>
    </div>
  </div>

            
                
  <div class="row">
    <div class="col-lg-12 col-xs-12">
      <div class="code-title">Get the activated conda environment inside the container</div>
    </div>
    <div class="col-lg-12 col-xs-12">
      <div class="code-container">
	<!-- <pre class="language-markup"><code>docker run -it kipoi/kipoi-docker:sharedpy3keras2</code></pre> -->
	<pre><code class="docker">docker run -it kipoi/kipoi-docker:sharedpy3keras2</code></pre>
	<div class="code-toolbar"><a data-toggle="tooltip" data-placement="top" data-html="true" title="Copy to clipboard!" class="btn btn-sm btn-info copy-to-clipboard">Copy</a></div>
      </div>
    </div>
  </div>

            
                
  <div class="row">
    <div class="col-lg-12 col-xs-12">
      <div class="code-title">Test the model</div>
    </div>
    <div class="col-lg-12 col-xs-12">
      <div class="code-container">
	<!-- <pre class="language-markup"><code>docker run kipoi/kipoi-docker:sharedpy3keras2 kipoi test DeepBind/Mus_musculus/TF/D00076.001_PBM_Sp140 --source=kipoi</code></pre> -->
	<pre><code class="docker">docker run kipoi/kipoi-docker:sharedpy3keras2 kipoi test DeepBind/Mus_musculus/TF/D00076.001_PBM_Sp140 --source=kipoi</code></pre>
	<div class="code-toolbar"><a data-toggle="tooltip" data-placement="top" data-html="true" title="Copy to clipboard!" class="btn btn-sm btn-info copy-to-clipboard">Copy</a></div>
      </div>
    </div>
  </div>

            
                
  <div class="row">
    <div class="col-lg-12 col-xs-12">
      <div class="code-title">Make prediction for custom files directly</div>
    </div>
    <div class="col-lg-12 col-xs-12">
      <div class="code-container">
	<!-- <pre class="language-markup"><code># Create an example directory containing the data
mkdir -p $PWD/kipoi-example 
# You can replace $PWD/kipoi-example with a different absolute path containing the data 
docker run -v $PWD/kipoi-example:/app/ kipoi/kipoi-docker:sharedpy3keras2 \
kipoi get-example DeepBind/Mus_musculus/TF/D00076.001_PBM_Sp140 -o /app/example 
docker run -v $PWD/kipoi-example:/app/ kipoi/kipoi-docker:sharedpy3keras2 \
kipoi predict DeepBind/Mus_musculus/TF/D00076.001_PBM_Sp140 \
--dataloader_args=&#39;{&#39;intervals_file&#39;: &#39;/app/example/intervals_file&#39;, &#39;fasta_file&#39;: &#39;/app/example/fasta_file&#39;}&#39; \
-o &#39;/app/DeepBind_Mus_musculus_TF_D00076.001_PBM_Sp140.example_pred.tsv&#39; 
# check the results
head $PWD/kipoi-example/DeepBind_Mus_musculus_TF_D00076.001_PBM_Sp140.example_pred.tsv
</code></pre> -->
	<pre><code class="docker"># Create an example directory containing the data
mkdir -p $PWD/kipoi-example 
# You can replace $PWD/kipoi-example with a different absolute path containing the data 
docker run -v $PWD/kipoi-example:/app/ kipoi/kipoi-docker:sharedpy3keras2 \
kipoi get-example DeepBind/Mus_musculus/TF/D00076.001_PBM_Sp140 -o /app/example 
docker run -v $PWD/kipoi-example:/app/ kipoi/kipoi-docker:sharedpy3keras2 \
kipoi predict DeepBind/Mus_musculus/TF/D00076.001_PBM_Sp140 \
--dataloader_args=&#39;{&#39;intervals_file&#39;: &#39;/app/example/intervals_file&#39;, &#39;fasta_file&#39;: &#39;/app/example/fasta_file&#39;}&#39; \
-o &#39;/app/DeepBind_Mus_musculus_TF_D00076.001_PBM_Sp140.example_pred.tsv&#39; 
# check the results
head $PWD/kipoi-example/DeepBind_Mus_musculus_TF_D00076.001_PBM_Sp140.example_pred.tsv
</code></pre>
	<div class="code-toolbar"><a data-toggle="tooltip" data-placement="top" data-html="true" title="Copy to clipboard!" class="btn btn-sm btn-info copy-to-clipboard">Copy</a></div>
      </div>
    </div>
  </div>

            
        </div>
    </div>

    <!-- singularity code snippet tab -->
    <div class="tab-pane" id="singularity" role="tabpanel">
        <div class="border border-top-0 code-snippet-container">
            
                
  <div class="row">
    <div class="col-lg-12 col-xs-12">
      <div class="code-title">Install apptainer</div>
    </div>
    <div class="col-lg-12 col-xs-12">
      <div class="code-container">
	<!-- <pre class="language-markup"><code>https://apptainer.org/docs/user/main/quick_start.html#quick-installation-steps</code></pre> -->
	<pre><code class="singularity">https://apptainer.org/docs/user/main/quick_start.html#quick-installation-steps</code></pre>
	<div class="code-toolbar"><a data-toggle="tooltip" data-placement="top" data-html="true" title="Copy to clipboard!" class="btn btn-sm btn-info copy-to-clipboard">Copy</a></div>
      </div>
    </div>
  </div>

            
                
  <div class="row">
    <div class="col-lg-12 col-xs-12">
      <div class="code-title">Make prediction for custom files directly</div>
    </div>
    <div class="col-lg-12 col-xs-12">
      <div class="code-container">
	<!-- <pre class="language-markup"><code>kipoi get-example DeepBind/Mus_musculus/TF/D00076.001_PBM_Sp140 -o example
kipoi predict DeepBind/Mus_musculus/TF/D00076.001_PBM_Sp140 \
--dataloader_args=&#39;{&#34;intervals_file&#34;: &#34;example/intervals_file&#34;, &#34;fasta_file&#34;: &#34;example/fasta_file&#34;}&#39; \
-o &#39;DeepBind_Mus_musculus_TF_D00076.001_PBM_Sp140.example_pred.tsv&#39; \
--singularity 
# check the results
head DeepBind_Mus_musculus_TF_D00076.001_PBM_Sp140.example_pred.tsv
</code></pre> -->
	<pre><code class="singularity">kipoi get-example DeepBind/Mus_musculus/TF/D00076.001_PBM_Sp140 -o example
kipoi predict DeepBind/Mus_musculus/TF/D00076.001_PBM_Sp140 \
--dataloader_args=&#39;{&#34;intervals_file&#34;: &#34;example/intervals_file&#34;, &#34;fasta_file&#34;: &#34;example/fasta_file&#34;}&#39; \
-o &#39;DeepBind_Mus_musculus_TF_D00076.001_PBM_Sp140.example_pred.tsv&#39; \
--singularity 
# check the results
head DeepBind_Mus_musculus_TF_D00076.001_PBM_Sp140.example_pred.tsv
</code></pre>
	<div class="code-toolbar"><a data-toggle="tooltip" data-placement="top" data-html="true" title="Copy to clipboard!" class="btn btn-sm btn-info copy-to-clipboard">Copy</a></div>
      </div>
    </div>
  </div>

            
        </div>
    </div>

</div>
                        </div>
                    </div>
                    
                    <hr class="w-100 clearfix">

                    <div class="row">
                        <div class="col-lg-5 col-md-12">
                            <!-- Model schema -->
                            <div class="card">
    <div class="card-header primary-color white-text">
        Schema
    </div>
    <div class="card-body">
        <div class="row">
            <div class="col-lg-6 col-md-6 col-sm-12 col-xs-12">
                <h4 class="card-title mt-2 mb-3">Inputs</h4>
                
  <p>Single numpy array</p>
  
    <p><strong>Name</strong>: seq</p>
    <p>
      &nbsp;&nbsp;&nbsp;&nbsp;<strong>Shape</strong>: (101, 4)&nbsp;<a data-toggle="tooltip" data-placement="top" data-html="true" title="Numpy array shape for a single data point (i.e. batch dimension excluded)."><i class="fa fa-question-circle blue-grey-text" style="font-size: 0.9em;" aria-hidden="true"></i></a></p>
    <p>&nbsp;&nbsp;&nbsp;&nbsp;<strong>Doc</strong>: DNA sequence</p>


            </div>
            <hr class="w-100 clearfix d-md-none">
            <div class="col-lg-6 col-md-6 col-sm-12 col-xs-12">
                <h4 class="card-title mt-2 mb-3">Targets</h4>
                
  <p>Single numpy array</p>
  
    <p><strong>Name</strong>: binding_prob</p>
    <p>
      &nbsp;&nbsp;&nbsp;&nbsp;<strong>Shape</strong>: (1,)&nbsp;<a data-toggle="tooltip" data-placement="top" data-html="true" title="Numpy array shape for a single data point (i.e. batch dimension excluded)."><i class="fa fa-question-circle blue-grey-text" style="font-size: 0.9em;" aria-hidden="true"></i></a></p>
    <p>&nbsp;&nbsp;&nbsp;&nbsp;<strong>Doc</strong>: Protein binding probability</p>


            </div>
        </div>
    </div>
</div>
                        </div>
                        <hr class="w-100 clearfix d-lg-none">
                        
                            <div class="col-lg-7 col-md-12">
                                <!-- Dataloader details -->
                                <div class="card">
        <div class="card-header warning-color-dark white-text clearfix">
            Dataloader
        </div>
        <div class="card-body">
            <div class="row">
              <div class="col-lg-5">
		<!-- TODO - ifelse  -->
		
  		  <p><strong>Defined as:</strong> kipoiseq.dataloaders.SeqIntervalDl</p>
		
                  <p><strong>Doc:</strong> Dataloader for a combination of fasta and tab-delimited input files such as bed files. The dataloader extracts regions from the fasta file as defined in the tab-delimited `intervals_file` and converts them into one-hot encoded format. Returned sequences are of the type np.array with the shape inferred from the arguments: `alphabet_axis` and `dummy_axis`.
</p>
                  <p><strong>Authors:</strong> 
  
    
      
  Ziga Avsec
  
    <a href="https://github.com/avsecz" class="github">
      <i class="fa fa-github" aria-hidden="true"></i>
    </a>
  
  

    
    , 
  
    
      
  Roman Kreuzhuber
  
    <a href="https://github.com/krrome" class="github">
      <i class="fa fa-github" aria-hidden="true"></i>
    </a>
  
  

    
    
  
</p>
                  <p><strong>Type:</strong> Dataset</p>
                  <p><strong>License:</strong> MIT</p>

                </div>
                <hr class="w-100 clearfix d-lg-none">
                <div class="col-lg-7">
                  <h4 class="card-title mt-2 mb-3">Arguments</h4>
                  
  <div>
    
      <p>
	<strong>intervals_file</strong> :
	bed3+&lt;columns&gt; file path containing intervals + (optionally) labels
      </p>
    
      <p>
	<strong>fasta_file</strong> :
	Reference genome FASTA file path.
      </p>
    
      <p>
	<strong>num_chr_fasta</strong> (optional):
	True, the the dataloader will make sure that the chromosomes don&#39;t start with chr.
      </p>
    
      <p>
	<strong>label_dtype</strong> (optional):
	None, datatype of the task labels taken from the intervals_file. Example: str, int, float, np.float32
      </p>
    
      <p>
	<strong>use_strand</strong> (optional):
	reverse-complement fasta sequence if bed file defines negative strand. Requires a bed6 file
      </p>
    
      <p>
	<strong>ignore_targets</strong> (optional):
	if True, don&#39;t return any target variables
      </p>
    
  </div>

                </div>
            </div>
        </div>
</div>
    
                            </div>
                        
                    </div>
                
                    <hr class="w-100 clearfix">
                    
                    <!-- Model and dataloader dependencies -->
                    <div class="row">
                        <div class="col-lg-12 col-md-12 col-xs-12">
                            <div class="row">
                                <div class="col-lg-5 col-md-5 col-sm-12 col-xs-12">
    <div class="card">
        <div class="card-header">
            Model dependencies
        </div>
        <div class="card-body">
            <div class="row">
                <div class="col-lg-6 col-md-6 col-sm-6 col-xs-12">
                    <strong>conda</strong>:<br>
                    
  <ul class="dependencies">
    
      <li>h5py=2.10.0<br></li>
    
      <li>tensorflow=2.7.0<br></li>
    
      <li>keras=2.7.0<br></li>
    
      <li>python=3.7<br></li>
    
      <li>pysam=0.18.0<br></li>
    
      <li>pip=20.2.4</li>
    
</u>

                </div>
                <hr class="w-100 clearfix d-sm-none">
                <div class="col-lg-6 col-md-6 col-sm-6 col-xs-12">
                    <strong>pip</strong>:<br>
                    
  <ul class="dependencies">
    
</u>

                </div>
            </div>
        </div>
    </div>
</div>
<hr class="w-100 clearfix d-md-none">
<div class="col-lg-7 col-md-7 col-sm-12 col-xs-12">
    <div class="card">
        <div class="card-header">
            Dataloader dependencies
        </div>
        <div class="card-body">
            <div class="row">
                
                    <div class="col-lg-6 col-md-6 col-sm-6 col-xs-12">
                        <strong>conda</strong>:<br>
                        
  <ul class="dependencies">
    
      <li>bioconda::pybedtools<br></li>
    
      <li>bioconda::pyfaidx<br></li>
    
      <li>bioconda::pyranges<br></li>
    
      <li>numpy<br></li>
    
      <li>pandas</li>
    
</u>

                    </div>
                    <hr class="w-100 clearfix d-sm-none">
                    <div class="col-lg-6 col-md-6 col-sm-6 col-xs-12">
                        <strong>pip</strong>:<br>
                        
  <ul class="dependencies">
    
      <li>kipoiseq</li>
    
</u>

                    </div>
                
            </div>
        </div>
    </div>
</div>
                            </div>
                        </div>
                    </div>
                    <hr class="w-100 clearfix">
                    <div class="row">
                        <div class="col-lg-12 col-md-12 col-xs-12">
                          <div id="disqus_thread"></div>
                        </div>
                    </div>
                </div>
        </section>
        <!--/Section: model-details content-->                        
    </div>

    </section>
    
    <footer class="page-footer center-on-small-only unique-color">
    <!--Footer links-->
    <div class="container-fluid">
        <div class="row">
            <!--First column-->
            <div class="col-lg-3 col-md-6 ml-auto">
                <h5 class="title mb-3"><strong>About KIPOI project</strong></h5>
                <p>Kipoi is an API and a repository of ready-to-use trained models for genomics.</p>
            </div>
            <!--/.First column-->
            <hr class="w-100 clearfix d-md-none">
            <!--Second column-->
            <div class="col-lg-2 col-md-6 ml-auto">
              <h5 class="title mb-3"><strong>Useful links</strong></h5>
              <ul>
                <li>
                  <a href="https://github.com/kipoi/kipoi">API repository <i class="fa fa-github" aria-hidden="true"></i></a>
                </li>
                <li>
                  <a href="https://github.com/kipoi/models">Model repository <i class="fa fa-github" aria-hidden="true"></i></a>
                </li>
                <li>
                  <a href="https://github.com/kipoi/website">Website repository <i class="fa fa-github" aria-hidden="true"></i></a>
                </li>
              </ul>
            </div>
            <!--/.Second column-->
            <hr class="w-100 clearfix d-lg-none">
            <!--Third column-->
            <div class="col-lg-4 col-md-12 ml-auto text-center">
                <ul>
                    <li>
                        <h5>Like the project?</h5>
                    </li>
                    <li>
                        <a target="_blank" href="https://twitter.com/KipoiZoo" class="btn btn-primary waves-effect waves-light" rel="nofollow">Tweet us!</a>
                    </li>
                </ul>
            </div>
            <!--/.Third column-->
        </div>
    </div>
    <!--/.Footer links-->

    <!--Copyright-->
    <div class="footer-copyright">
        <div class="containter-fluid">
            © 2022 The Kipoi team, website developed by 
            <a href="https://github.com/zupan" class="Nejc Zupan github">
                Nejc Zupan
                <i class="fa fa-github" aria-hidden="true"></i>
            </a>
        </div>
    </div>
    <!--/.Copyright-->
</footer>

    <!-- SCRIPTS -->
<!-- JQuery -->
<script type="text/javascript" src="/static/js/jquery-3.2.1.min.js"></script>
<!-- Bootstrap tooltips -->
<script type="text/javascript" src="/static/js/popper.min.js"></script>
<!-- Bootstrap core JavaScript -->
<script type="text/javascript" src="/static/js/bootstrap.min.js"></script>
<!-- MDB core JavaScript -->
<script type="text/javascript" src="/static/js/mdb.min.js"></script>

<!-- DataTables -->
<script type="text/javascript" src="/static/js/jquery.dataTables.min.js"></script>
<script type="text/javascript" src="/static/js/dataTables.material.min.js"></script>

<!-- Highlight -->
<script type="text/javascript" src="/static/js/highlight.pack.js"></script>

<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-115025493-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-115025493-1');
</script>
    
    
    <script language="JavaScript">
      // Enable code highlighting
      hljs.initHighlightingOnLoad();

      $(document).ready(function() {
        $('[data-toggle="tooltip"]').tooltip();

        $('.copy-to-clipboard').on('click', function() {
          /* Get the code field */
          var copyCode = $(this).closest('.code-container').find('code').text();
          var $temp = $("<textarea>");
          $("body").append($temp);
          $temp.val(copyCode).select();
          document.execCommand("copy");
          $temp.remove();
        });
      });

      /**
       *  RECOMMENDED CONFIGURATION VARIABLES: EDIT AND UNCOMMENT THE SECTION BELOW TO INSERT DYNAMIC VALUES FROM YOUR PLATFORM OR CMS.
       *  LEARN WHY DEFINING THESE VARIABLES IS IMPORTANT: https://disqus.com/admin/universalcode/#configuration-variables*/
      
      var disqus_config = function () {
        this.page.url = "http://localhost/models/DeepBind/Mus_musculus/TF/D00076.001_PBM_Sp140/";  // Replace PAGE_URL with your page's canonical URL variable
        this.page.identifier = "/models/DeepBind/Mus_musculus/TF/D00076.001_PBM_Sp140/"; // Replace PAGE_IDENTIFIER with your page's unique identifier variable
      };
      
      (function() { // DON'T EDIT BELOW THIS LINE
        var d = document, s = d.createElement('script');
        s.src = 'https://kipoi-org.disqus.com/embed.js';
        s.setAttribute('data-timestamp', +new Date());
        (d.head || d.body).appendChild(s);
      })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    

</body>

</html>