<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  
  <link rel="shortcut icon" href="../../img/favicon.ico">
  <title>MutationMap - Kipoi-veff</title>
  <link href='https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700' rel='stylesheet' type='text/css'>

  <link rel="stylesheet" href="../../css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../../css/theme_extra.css" type="text/css" />
  <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css">
  
  <script>
    // Current page data
    var mkdocs_page_name = "MutationMap";
    var mkdocs_page_input_path = "api/MutationMap.md";
    var mkdocs_page_url = null;
  </script>
  
  <script src="../../js/jquery-2.1.1.min.js" defer></script>
  <script src="../../js/modernizr-2.8.3.min.js" defer></script>
  <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
  <script>hljs.initHighlightingOnLoad();</script> 
  
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
      <div class="wy-side-nav-search">
        <a href="../.." class="icon icon-home"> Kipoi-veff</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
	<ul class="current">
	  
          
            <li class="toctree-l1">
		
    <a class="" href="../..">Home</a>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../../overview/">Overview</a>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Using</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="../../using/">Functions</a>
                </li>
                <li class="">
                    
    <a class="" href="../../variant_effect_pred/">Variant effect prediction</a>
                </li>
                <li class="">
                    
    <a class="" href="../../mutation_map/">Mutation maps</a>
                </li>
    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Tutorials</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="../../tutorials/variant_effect_prediction_simple/">Simple</a>
                </li>
                <li class="">
                    
    <a class="" href="../../tutorials/variant_effect_prediction/">Advanced</a>
                </li>
                <li class="">
                    
    <a class="" href="../../tutorials/mutation_map/">Mutation map</a>
                </li>
    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Api</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="../score_variants/">score_variants</a>
                </li>
                <li class="">
                    
    <a class="" href="../predict_snvs/">predict_snvs</a>
                </li>
                <li class="">
                    
    <a class="" href="../scores/">Available scores</a>
                </li>
                <li class=" current">
                    
    <a class="current" href="./">MutationMap</a>
    <ul class="subnav">
            
    </ul>
                </li>
                <li class="">
                    
    <a class="" href="../MutationMapPlotter/">MutationMapPlotter</a>
                </li>
                <li class="">
                    
    <a class="" href="../KipoiVCFParser/">KipoiVCFParser</a>
                </li>
    </ul>
	    </li>
          
        </ul>
      </div>
      &nbsp;
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="../..">Kipoi-veff</a>
      </nav>

      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="../..">Docs</a> &raquo;</li>
    
      
        
          <li>Api &raquo;</li>
        
      
    
    <li>MutationMap</li>
    <li class="wy-breadcrumbs-aside">
      
        <a href="http://github.com/kipoi/kipoi-veff/edit/master/docs/api/MutationMap.md"
          class="icon icon-github"> Edit on GitHub</a>
      
    </li>
  </ul>
  <hr/>
</div>
          <div role="main">
            <div class="section">
              
                <h1 id="kipoi_veff.MutationMap">MutationMap</h1>

<pre><code class="python">MutationMap(self, model, dataloader, dataloader_args=None, use_dataloader_example_data=False)
</code></pre>

<p>Generate mutation map</p>
<p>Prediction of effects of every base at every position of datalaoder input sequences. The regions for which
the effect scores will be calculated are primarily defined by the dataloader input.</p>
<p><strong>Arguments</strong></p>
<ul>
<li><strong>model</strong>: A kipoi model handle generated by e.g.: kipoi.get_model()</li>
<li><strong>dataloader</strong>: Dataloader factory generated by e.g.: kipoi.get_dataloader_factory()</li>
<li><strong>dataloader_args</strong>: arguments passed on to the dataloader for sequence generation, arguments
        mentioned in dataloader.yaml &gt; postprocessing &gt; variant_effects &gt; bed_input will be overwritten
        by the methods here.</li>
<li><strong>use_dataloader_example_data</strong>: Fill out the missing dataloader arguments with the example values given in
        the dataloader.yaml.</li>
</ul>
<h2 id="kipoi_veff.MutationMap.query_region">query_region</h2>

<pre><code class="python">MutationMap.query_region(self, chrom, start, end, seq_length=None, scores=['logit_ref', 'logit_alt', 'ref', 'alt', 'logit', 'diff'], score_kwargs=None, **kwargs)
</code></pre>

<p>Generate mutation map</p>
<p>Prediction of effects of every base at every position of datalaoder input sequences. The regions for which
the effect scores will be calculated are primarily defined by the dataloader input. If the dataloader accepts
<code>bed</code> file inputs then this file will be overwritten with regions defined here of length <code>seq_length</code> or
the model input sequence length. If that is not available all datalaoder-generated regions that overlap the
region defined here will be investigated. Effect scores are returned as <code>MutationMapPlotter</code> object which can
be saved to  an hdf5 file and used for plotting. It is important to mention that the order of the scored
sequences is the order in which the dataloader has produced data input - intersected with the region defined
here.</p>
<p><strong>Arguments</strong></p>
<ul>
<li><strong>chrom</strong>: Chrosome of region of interest. Assembly is defined by the dataload arguments.</li>
<li><strong>start</strong>: Start of region of interest. Assembly is defined by the dataload arguments.</li>
<li><strong>end</strong>: End of region of interest. Assembly is defined by the dataload arguments.</li>
<li><strong>seq_length</strong>: Optional argument of model sequence length to use if model accepts variable input
    sequence length.</li>
<li><strong>scores</strong>: list of score names to compute. See kipoi_veff.scores</li>
<li><strong>score_kwargs</strong>: optional, list of kwargs that corresponds to the entries in score.</li>
</ul>
<p><strong>Returns</strong></p>
<p><code>MutationMapPlotter</code>: object containing variant scores.</p>
<h2 id="kipoi_veff.MutationMap.query_bed">query_bed</h2>

<pre><code class="python">MutationMap.query_bed(self, bed_fpath, seq_length=None, scores=['logit_ref', 'logit_alt', 'ref', 'alt', 'logit', 'diff'], score_kwargs=None, **kwargs)
</code></pre>

<p>Generate mutation map</p>
<p>Prediction of effects of every base at every position of datalaoder input sequences. The regions for which
the effect scores will be calculated are primarily defined by the dataloader input. If the dataloader accepts
<code>bed</code> file inputs then this file will be overwritten with regions defined in <code>bed_fpath</code> of length
<code>seq_length</code> or the model input sequence length. If that is not available all datalaoder-generated
regions that overlap the region defined here will be investigated. Effect scores are returned as
<code>MutationMapPlotter</code> object which can be saved to  an hdf5 file and used for plotting. It is important to
mention that the order of the scored sequences is the  order in which the dataloader has produced data
input - intersected with <code>bed_fpath</code>.</p>
<p><strong>Arguments</strong></p>
<ul>
<li><strong>bed_fpath</strong>: Only genomic regions overlapping regions in this bed file will be evaluated. If
        the dataloader accepts bed file input then the dataloader bed input file will be overwritten with
        regions based this (<code>bed_fpath</code>) bed file. Assembly is defined by the dataload arguments.</li>
<li><strong>seq_length</strong>: Optional argument of model sequence length to use if model accepts variable input
        sequence length.</li>
<li><strong>scores</strong>: list of score names to compute. See kipoi_veff.scores</li>
<li><strong>score_kwargs</strong>: optional, list of kwargs that corresponds to the entries in score.</li>
</ul>
<p><strong>Returns</strong></p>
<p><code>MutationMapPlotter</code>: object containing variant scores.</p>
<h2 id="kipoi_veff.MutationMap.query_vcf">query_vcf</h2>

<pre><code class="python">MutationMap.query_vcf(self, vcf_fpath, seq_length=None, scores=['logit_ref', 'logit_alt', 'ref', 'alt', 'logit', 'diff'], score_kwargs=None, var_centered_regions=True, **kwargs)
</code></pre>

<p>Generate mutation map</p>
<p>Prediction of effects of every base at every position of datalaoder input sequences. The regions for which
the effect scores will be calculated are primarily defined by the dataloader input. If the dataloader accepts
<code>bed</code> file inputs then this file will be overwritten with regions generaten from the SNVs in <code>vcf_fpath</code>in a
variant-centered fashion. Sequence length is defined by <code>seq_length</code> or the model input sequence length.
If the datalaoder does not have a <code>bed</code> file input all datalaoder-generated regions that overlap SNVs
here will be investigated. Effect scores are returned as <code>MutationMapPlotter</code> object which can be saved to
an hdf5 file and used for plotting. It is important to mention that the order of the scored sequences is the
order in which the dataloader has produced data input - intersected with <code>vcf_fpath</code>.</p>
<p><strong>Arguments</strong></p>
<ul>
<li><strong>vcf_fpath</strong>: Only genomic regions overlapping the variants in this VCF will be evaluated.
        Variants defined here will be highlighted in mutation map plots. Only SNVs will be used. If
        vcf_to_region is defined and the dataloader accepts bed file input then the dataloader bed input
        file will be overwritten with regions based on variant positions of this VCF.</li>
<li><strong>seq_length</strong>: Optional argument of model sequence length to use if model accepts variable input
        sequence length.</li>
<li><strong>var_centered_regions</strong>: Generate variant-centered regions if the model accepts that. If a custom
        <code>vcf_to_region</code> should be used then this can be set explicitly in the kwargs.</li>
<li><strong>scores</strong>: list of score names to compute. See kipoi_veff.scores</li>
<li><strong>score_kwargs</strong>: optional, list of kwargs that corresponds to the entries in score.</li>
</ul>
<p><strong>Returns</strong></p>
<p><code>MutationMapPlotter</code>: object containing variant scores.</p>
              
            </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../MutationMapPlotter/" class="btn btn-neutral float-right" title="MutationMapPlotter">Next <span class="icon icon-circle-arrow-right"></span></a>
      
      
        <a href="../scores/" class="btn btn-neutral" title="Available scores"><span class="icon icon-circle-arrow-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
    
  </div>

  Built with <a href="http://www.mkdocs.org">MkDocs</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
      
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" style="cursor: pointer">
    <span class="rst-current-version" data-toggle="rst-current-version">
      
          <a href="http://github.com/kipoi/kipoi-veff/" class="fa fa-github" style="float: left; color: #fcfcfc"> GitHub</a>
      
      
        <span><a href="../scores/" style="color: #fcfcfc;">&laquo; Previous</a></span>
      
      
        <span style="margin-left: 15px"><a href="../MutationMapPlotter/" style="color: #fcfcfc">Next &raquo;</a></span>
      
    </span>
</div>
    <script>var base_url = '../..';</script>
    <script src="../../js/theme.js" defer></script>
      <script src="../../search/main.js" defer></script>

</body>
</html>
