
<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
    <meta http-equiv="x-ua-compatible" content="ie=edge">
    <meta name="author" content="Nejc Zupan">
    <link rel="icon" type="image/png" href="/static/img/favicon.png">
    <title>Kipoi</title>
    <!-- Font Awesome -->
<link rel="stylesheet" type="text/css" href="/static/css/font-awesome.min.css">
<!-- Bootstrap core CSS -->
<link rel="stylesheet" type="text/css" href="/static/css/bootstrap.min.css">
<!-- Material Design Bootstrap -->
<link rel="stylesheet" type="text/css" href="/static/css/mdb.min.css">
<!-- Material Design Lite -->
<link rel="stylesheet" type="text/css" href="/static/css/material.min.css">
<!-- DataTables Material Design -->    
<link rel="stylesheet" type="text/css" href="/static/css/dataTables.material.css"/>
<!-- Highlight -->
<link rel="stylesheet" type="text/css" href="/static/css/custom_highlight_github.css"/>

<!-- Custom styles -->
<link rel="stylesheet" type="text/css" href="/static/css/style.css?v=150318">
</head>

<body>
    <section class="mb-3">
        
<!--Navbar-->

<nav class="navbar fixed-top navbar-expand-md navbar-dark unique-color-kipoi">

    <!-- Navbar brand -->
    <span>
      <a class="nav-title-link navbar-brand-nopad text-hide left-margin" href="/"
	 ><img src="/static/img/nav_bar_logo_v3.png" height="50px" alt="Kipoi"/></a>
    </span>

    <!-- Collapse button -->
    <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarNav1" aria-controls="navbarNav1" aria-expanded="false" aria-label="Toggle navigation"><span class="navbar-toggler-icon"></span></button>
    
    <!-- Collapsible content -->
    <div class="collapse navbar-collapse text-center" id="navbarNav1">
        
        <!-- Links -->
        <ul class="navbar-nav mr-auto text-center">
	  <!-- Group list --> 
            <li class="nav-item">
              <a id="components-related-1" class="nav-button btn btn-outline-primary navbar-dark nabar-button btn-active" href="/groups/">Models</a>
            </li>
            <!-- About --> 
            <li class="nav-item">
              <a id="components-related-1" class="nav-button btn btn-outline-primary navbar-dark nabar-button btn-active" href="/about/">About</a>
            </li>
	    <!-- Docs --> 
            <li class="nav-item">
              <a id="components-related-3" class="nav-button btn btn-outline-primary navbar-dark nabar-button" href="/docs/">Docs</a>
            </li>
        </ul>

        <!-- Search form -->
        
          <form class="navbar-nav form-inline promenent-search rounded">
            <div class="text-nowrap" style="width: 100%;">
              <label for="search-bar"><i class="fa fa-search fa-fw px-3" aria-hidden="true"></i></label>
              <input id="search-bar" class="form-control model-search" type="text" placeholder="Search" aria-label="Search" autofocus/>
            </div>
          </form>
        

      
        <ul class="navbar-nav ml-auto text-center d-md-none d-lg-flex">
      
          <li class="nav-item">
            <a id="components-related-4" class="text-nowrap nav-link waves-effect waves-light" href="https://www.biorxiv.org/content/early/2018/07/24/375345.full.pdf">White paper<i class="fa fa-file-text-o" aria-hidden="true"></i></a>
          </li>
          <li class="nav-item">
            <a id="components-related-4" class="text-nowrap nav-link waves-effect waves-light" href="https://github.com/kipoi/models">Model repository<i class="fa fa-github" aria-hidden="true"></i></a>
          </li>
          <li class="nav-item">
            <a id="components-related-5" class="text-nowrap nav-link waves-effect waves-light" href="https://github.com/kipoi/kipoi">API repository<i class="fa fa-github" aria-hidden="true"></i></a>
          </li>
        </ul>
        
    </div>
</nav>
    </section>

    <section class="main-content">
        
<div class="row" style="margin-right: 0; margin-left: 0;">
    <div class="col-xl-12">
        <!--Section: model-list content-->
        <section class="model-list">
            <table id="model-list-table" class="mdl-data-table table-responsive-xl" cellspacing="0" style="width:100%">
    <thead>
        <tr>
            <th class="text-left">Model name</th>
            <th>Version</th>
            <th>Authors</th>
            <th>Contributed by</th>
            <th>Type</th>
            <th>Cite as</th>
            <th>License</th>
            <th>Training procedure</th>
        </tr>
    </thead>
    <tfoot>
        <tr>
            <th class="text-left">Model name</th>
            <th>Version</th>
            <th>Authors</th>
            <th>Contributed by</th>
            <th>Type</th>
            <th>Cite as</th>
            <th>License</th>
            <th>Training procedure</th>
        </tr>
    </tfoot>
    <tbody>
        
        <tr>
            <td class="text-left">
                <a class="" href="/models/MMSplice/pathogenicity">
                MMSplice/pathogenicity
                </a>
            </td>
            <td>0.1</td>
            <td>
              
  
     
      
	
  Jun Cheng
  
    <a href="https://github.com/s6juncheng" class="github">
      <i class="fa fa-github" aria-hidden="true"></i>
    </a>
  
  

      
    
    
  

            </td>
            <td>
              
  
     
      
	
  Jun Cheng
  
    <a href="https://github.com/s6juncheng" class="github">
      <i class="fa fa-github" aria-hidden="true"></i>
    </a>
  
  

      
    
    
  

            </td>
            <td>custom</td>
            <td>
                
  

            </td>
            <td>
                
  
    MIT
  

            </td>
            <td>
                
  

            </td>
        </tr>
        
        <tr>
            <td class="text-left">
                <a class="" href="/models/MMSplice/splicingEfficiency">
                MMSplice/splicingEfficiency
                </a>
            </td>
            <td>0.1</td>
            <td>
              
  
     
      
	
  Jun Cheng
  
    <a href="https://github.com/s6juncheng" class="github">
      <i class="fa fa-github" aria-hidden="true"></i>
    </a>
  
  

      
    
    
  

            </td>
            <td>
              
  
     
      
	
  Jun Cheng
  
    <a href="https://github.com/s6juncheng" class="github">
      <i class="fa fa-github" aria-hidden="true"></i>
    </a>
  
  

      
    
    
  

            </td>
            <td>custom</td>
            <td>
                
  

            </td>
            <td>
                
  
    MIT
  

            </td>
            <td>
                
  

            </td>
        </tr>
        
        <tr>
            <td class="text-left">
                <a class="" href="/models/MMSplice/modularPredictions">
                MMSplice/modularPredictions
                </a>
            </td>
            <td>0.1</td>
            <td>
              
  
     
      
	
  Jun Cheng
  
    <a href="https://github.com/s6juncheng" class="github">
      <i class="fa fa-github" aria-hidden="true"></i>
    </a>
  
  

      
    
    
  

            </td>
            <td>
              
  
     
      
	
  Jun Cheng
  
    <a href="https://github.com/s6juncheng" class="github">
      <i class="fa fa-github" aria-hidden="true"></i>
    </a>
  
  

      
    
    
  

            </td>
            <td>custom</td>
            <td>
                
  

            </td>
            <td>
                
  
    MIT
  

            </td>
            <td>
                
  

            </td>
        </tr>
        
        <tr>
            <td class="text-left">
                <a class="" href="/models/MMSplice/mtsplice">
                MMSplice/mtsplice
                </a>
            </td>
            <td>0.1</td>
            <td>
              
  
     
      
	
  Jun Cheng
  
    <a href="https://github.com/s6juncheng" class="github">
      <i class="fa fa-github" aria-hidden="true"></i>
    </a>
  
  

      
    
    
  

            </td>
            <td>
              
  
     
      
	
  Jun Cheng
  
    <a href="https://github.com/s6juncheng" class="github">
      <i class="fa fa-github" aria-hidden="true"></i>
    </a>
  
  

      
    
    
  

            </td>
            <td>custom</td>
            <td>
                
  

            </td>
            <td>
                
  
    MIT
  

            </td>
            <td>
                
  

            </td>
        </tr>
        
        <tr>
            <td class="text-left">
                <a class="" href="/models/MMSplice/deltaLogitPSI">
                MMSplice/deltaLogitPSI
                </a>
            </td>
            <td>0.1</td>
            <td>
              
  
     
      
	
  Jun Cheng
  
    <a href="https://github.com/s6juncheng" class="github">
      <i class="fa fa-github" aria-hidden="true"></i>
    </a>
  
  

      
    
    
  

            </td>
            <td>
              
  
     
      
	
  Jun Cheng
  
    <a href="https://github.com/s6juncheng" class="github">
      <i class="fa fa-github" aria-hidden="true"></i>
    </a>
  
  

      
    
    
  

            </td>
            <td>custom</td>
            <td>
                
  

            </td>
            <td>
                
  
    MIT
  

            </td>
            <td>
                
  

            </td>
        </tr>
        
    </tbody>
</table>
<div class="row" style="margin-right: 0; margin-left: 0;">
  <div class="col-lg-9 mx-auto readmecontent"><h2>Modular Modeling of Splicing (MMSplice)</h2>
<p>MMSplice predicts variant effect with 5 modules scoring exon, donor, acceptor, 3' intron and 5' intron. Modular predictions are combined with a linear model to predict $\Delta logit(\Psi)$ or with a logistic regression model to predict variant pathogenicity.</p>
<p>To score variants (including indels), we suggest to use primarily the <code>deltaLogitPSI</code> predictions.</p>
<h3>Quick start example code</h3>
<hr />
<p>Install <code>kipoi</code> as described in: <a href="http://kipoi.org/docs/#installation">http://kipoi.org/docs/#installation</a></p>
<ol>
<li>Create a new conda environment with all dependencies installed
<pre class="highlight"><code class="language-bash">kipoi env create MMSplice/deltaLogitPSI
source activate kipoi-MMSplice__deltaLogitPSI</code></pre></li>
<li>
<p>Install model dependencies into current environment
<pre class="highlight"><code class="language-bash">kipoi env install MMSplice/deltaLogitPSI</code></pre></p>
</li>
<li>
<p>Run deltaLogitPSI prediction on example files from the command line with <code>kipoi predict</code>. MMSplice takes three input files: a gtf annotation file, a reference sequence fasta file and a vcf file with variants.</p>
</li>
</ol>
<pre class="highlight"><code class="language-bash">cd ~/.kipoi/models/MMSplice/deltaLogitPSI
kipoi predict MMSplice/deltaLogitPSI \
  --dataloader_args='{'gtf_file': 'example_files/test.gtf', 'fasta_file': 'example_files/hg19.nochr.chr17.fa', 'vcf_file': 'example_files/test.vcf.gz'}' \  -o test.tsv
# check the results
head 'test.tsv'</code></pre>

<p>We also documented the usage with CLI, python and R <a href="http://kipoi.org/models/MMSplice/deltaLogitPSI/">here</a>. </p>
<h3>Prepare your own input files</h3>
<h4>1. Prepare annotation (gtf) file</h4>
<p>Standard human gene annotation file in GTF format can be downloaded from ensembl or gencode.
<code>MMSplice</code> can work directly with those files, however, some filtering is higly recommended.</p>
<ul>
<li>Filter for protein coding genes.</li>
<li>Filter out duplicated exons. The same exon can be annotated multiple times if it appears in multiple transcripts. 
  This will cause duplicated predictions.</li>
</ul>
<p>We provide a filtered version <a href="https://raw.githubusercontent.com/gagneurlab/MMSplice_paper/master/data/shared/Homo_sapiens.GRCh37.75.chr.uniq_exon.gtf.gz">here</a>. 
Note this version has chromosome names in the format <code>chr*</code>. You may need to remove them to match the chromosome names in your fasta file.</p>
<h4>2. Prepare variant (VCF) file</h4>
<p>A correctly formatted VCF file will work with <code>MMSplice</code>, however the following steps will make it less prone to false positives:</p>
<ul>
<li>Quality filtering. Low quality variants lead to unreliable predictions.</li>
<li>Avoid presenting multiple variants in one line by splitting them into multiple lines. Example code to do this:
  <pre class="highlight"><code class="language-bash">bcftools norm -m-both -o out.vcf in.vcf.gz</code></pre></li>
<li>Left-normalization. For instance, GGCA--&gt;GG is not left-normalized while GCA--&gt;G is. Details on unified representation of genetic variants see <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4481842/">Tan et al.</a>
  <pre class="highlight"><code class="language-bash">bcftools norm -f reference.fasta -o out.vcf in.vcf</code></pre></li>
</ul>
<h4>3. Prepare reference genome (fasta) file</h4>
<p>Human reference fasta file can be downloaded from ensembl/gencode. Make sure the chromosome names match with the GTF annotation file you use.</p>
<!-- We show example with `deltaLogitPSI` model, other models follow the same logic. Kipoi also documented the usage with CLI, python and R [here](http://kipoi.org/models/MMSplice/). Click on the model of interest, for instance, to predict [deltaLogitPsi](http://kipoi.org/models/MMSplice/deltaLogitPSI/). -->

<h2>Repository</h2>
<p>This repository hosts following models:</p>
<p><code>deltaLogitPSI</code>: predict $\Delta logit(\Psi)$. Returns one prediction per variant-exon pair. </p>
<p><code>pathogenicity</code>: predict variant pathogenicity. Returns one prediction per variant.</p>
<p><code>splicingEfficiency</code>: predict splicing efficiency changes. Returns one prediction per variant-exon pair. Splicing efficiency model were trained from MMSplice modules and exonic variants from MaPSy, thus only the predictions for exonic variants are calibrated. </p>
<p><code>modularPredictions</code>: the raw predictions from the five modules for reference sequence and alternative sequence. Returns a vector of length 10 for each variant-exon pair. </p>
<p><code>modules</code>: contains individual modules for exon, donor, acceptor, 3' intron and 5' intron.</p>
<p><code>deltaLogitPSI</code>, <code>splicingEfficiency</code> and <code>pathogenicity</code> differ by the last modular combination model. </p>
<h2>Publication</h2>
<p>Paper: Cheng et al. <a href="https://doi.org/10.1101/438986">https://doi.org/10.1101/438986</a></p>
<p>Package: <a href="https://github.com/gagneurlab/MMSplice">https://github.com/gagneurlab/MMSplice</a></p>
<p><img alt="MMSplice" src="https://raw.githubusercontent.com/kipoi/models/master/MMSplice/Model.png" /></p></div>
</div>
        </section>
        <!--/Section: model-list content-->                        
    </div>
    
</div>

    </section>
    
    <footer class="page-footer center-on-small-only unique-color">
    <!--Footer links-->
    <div class="container-fluid">
        <div class="row">
            <!--First column-->
            <div class="col-lg-3 col-md-6 ml-auto">
                <h5 class="title mb-3"><strong>About KIPOI project</strong></h5>
                <p>Kipoi is an API and a repository of ready-to-use trained models for genomics.</p>
            </div>
            <!--/.First column-->
            <hr class="w-100 clearfix d-md-none">
            <!--Second column-->
            <div class="col-lg-2 col-md-6 ml-auto">
              <h5 class="title mb-3"><strong>Useful links</strong></h5>
              <ul>
                <li>
                  <a href="https://github.com/kipoi/kipoi">API repository <i class="fa fa-github" aria-hidden="true"></i></a>
                </li>
                <li>
                  <a href="https://github.com/kipoi/models">Model repository <i class="fa fa-github" aria-hidden="true"></i></a>
                </li>
                <li>
                  <a href="https://github.com/kipoi/website">Website repository <i class="fa fa-github" aria-hidden="true"></i></a>
                </li>
              </ul>
            </div>
            <!--/.Second column-->
            <hr class="w-100 clearfix d-lg-none">
            <!--Third column-->
            <div class="col-lg-4 col-md-12 ml-auto text-center">
                <ul>
                    <li>
                        <h5>Like the project?</h5>
                    </li>
                    <li>
                        <a target="_blank" href="https://twitter.com/KipoiZoo" class="btn btn-primary waves-effect waves-light" rel="nofollow">Tweet us!</a>
                    </li>
                </ul>
            </div>
            <!--/.Third column-->
        </div>
    </div>
    <!--/.Footer links-->

    <!--Copyright-->
    <div class="footer-copyright">
        <div class="containter-fluid">
            Â© 2020 The Kipoi team, website developed by 
            <a href="https://github.com/zupan" class="Nejc Zupan github">
                Nejc Zupan
                <i class="fa fa-github" aria-hidden="true"></i>
            </a>
        </div>
    </div>
    <!--/.Copyright-->
</footer>

    <!-- SCRIPTS -->
<!-- JQuery -->
<script type="text/javascript" src="/static/js/jquery-3.2.1.min.js"></script>
<!-- Bootstrap tooltips -->
<script type="text/javascript" src="/static/js/popper.min.js"></script>
<!-- Bootstrap core JavaScript -->
<script type="text/javascript" src="/static/js/bootstrap.min.js"></script>
<!-- MDB core JavaScript -->
<script type="text/javascript" src="/static/js/mdb.min.js"></script>

<!-- DataTables -->
<script type="text/javascript" src="/static/js/jquery.dataTables.min.js"></script>
<script type="text/javascript" src="/static/js/dataTables.material.min.js"></script>

<!-- Highlight -->
<script type="text/javascript" src="/static/js/highlight.pack.js"></script>

<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-115025493-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-115025493-1');
</script>
    
    
    <script language="JavaScript">
        $(document).ready(function() {
            $('#model-list-table').DataTable({
                "lengthMenu": [ [25, 50, 100, -1], [25, 50, 100, "All"] ],
                "pageLength": 50,
                "dom": 'rtip'
            });
            $('.dt-table').css('width', '100%');

            oTable = $('#model-list-table').DataTable();
            $('#search-bar').keyup(function(){
                oTable.search($(this).val()).draw() ;
            })
            $('.dataTables_info').addClass('col-md-5 text-right mt-2');
            $('.dataTables_paginate').addClass('col-md-7 mt-4');
			$('.readmecontent img').css('max-width','100%')
        });
    </script>

</body>

</html>