<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  
  
  <link rel="shortcut icon" href="../img/favicon.ico">
  <title>Dataloaders - Kipoiseq</title>
  <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700" />

  <link rel="stylesheet" href="../css/theme.css" />
  <link rel="stylesheet" href="../css/theme_extra.css" />
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.5.0/styles/github.min.css" />
  
  <script>
    // Current page data
    var mkdocs_page_name = "Dataloaders";
    var mkdocs_page_input_path = "dataloaders.md";
    var mkdocs_page_url = null;
  </script>
  
  <script src="../js/jquery-2.1.1.min.js" defer></script>
  <script src="../js/modernizr-2.8.3.min.js" defer></script>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.5.0/highlight.min.js"></script>
  <script>hljs.initHighlightingOnLoad();</script> 
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
    <div class="wy-side-scroll">
      <div class="wy-side-nav-search">
        <a href=".." class="icon icon-home"> Kipoiseq</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../search.html" method="get">
      <input type="text" name="q" placeholder="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="..">Home</a>
                    </li>
                </ul>
                <ul class="current">
                    <li class="toctree-l1 current"><a class="reference internal current" href="./">Dataloaders</a>
    <ul class="current">
    </ul>
                    </li>
                </ul>
                <p class="caption"><span class="caption-text">Transforms</span></p>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../transforms/functional/">Functional</a>
                    </li>
                    <li class="toctree-l1"><a class="reference internal" href="../transforms/transforms/">Class-based</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../extractors/">Extractors</a>
                    </li>
                </ul>
      </div>
    </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="..">Kipoiseq</a>
      </nav>

      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="..">Docs</a> &raquo;</li>
    
      
    
    <li>Dataloaders</li>
    <li class="wy-breadcrumbs-aside">
      
        <a href="http://github.com/kipoi/kipoiseq/edit/master/docs/dataloaders.md"
          class="icon icon-github"> Edit on GitHub</a>
      
    </li>
  </ul>
  
  <hr/>
</div>

          <div role="main">
            <div class="section">
              
                <h2 id="beddataset">BedDataset</h2>
<pre><code class="language-python">BedDataset(self, tsv_file, label_dtype=None, bed_columns=3, num_chr=False, ambiguous_mask=None, incl_chromosomes=None, excl_chromosomes=None, ignore_targets=False)
</code></pre>
<p>Reads a tsv file in the following format:</p>
<pre><code>chr  start  stop  task1  task2 ...
</code></pre>
<p><strong>Arguments</strong></p>
<ul>
<li><strong>tsv_file</strong>: tsv file type</li>
<li><strong>bed_columns</strong>: number of columns corresponding to the bed file. All the columns
    after that will be parsed as targets</li>
<li><strong>num_chr</strong>: if specified, 'chr' in the chromosome name will be dropped</li>
<li><strong>label_dtype</strong>: specific data type for labels, Example: <code>float</code> or <code>np.float32</code></li>
<li><strong>ambiguous_mask</strong>: if specified, rows containing only ambiguous_mask values will be skipped</li>
<li><strong>incl_chromosomes</strong>: exclusive list of chromosome names to include in the final dataset.
    if not None, only these will be present in the dataset</li>
<li><strong>excl_chromosomes</strong>: list of chromosome names to omit from the dataset.</li>
<li><strong>ignore_targets</strong>: if True, target variables are ignored</li>
</ul>
<h2 id="stringseqintervaldl">StringSeqIntervalDl</h2>
<pre><code class="language-python">StringSeqIntervalDl(self, intervals_file, fasta_file, num_chr_fasta=False, label_dtype=None, auto_resize_len=None, use_strand=False, force_upper=True, ignore_targets=False)
</code></pre>
<p>Dataloader for a combination of fasta and tab-delimited input files such as bed files. The dataloader extracts regions from the fasta file as defined in the tab-delimited <code>intervals_file</code>. Returned sequences are of the type np.array([str]).</p>
<p><strong>Arguments</strong></p>
<ul>
<li><strong>intervals_file</strong>: bed3+<columns> file path containing intervals + (optionally) labels <a href="https://raw.githubusercontent.com/kipoi/kipoiseq/master/tests/data/intervals_51bp.tsv">example</a></li>
<li><strong>fasta_file</strong>: Reference genome FASTA file path. <a href="https://raw.githubusercontent.com/kipoi/kipoiseq/master/tests/data/hg38_chr22_32000000_32300000.fa">example</a></li>
<li><strong>num_chr_fasta</strong>: True, the the dataloader will make sure that the chromosomes don't start with chr.</li>
<li><strong>label_dtype</strong>: None, datatype of the task labels taken from the intervals_file. Example - str, int, float, np.float32</li>
<li><strong>auto_resize_len</strong>: None, required sequence length.</li>
<li><strong>use_strand</strong>: reverse-complement fasta sequence if bed file defines negative strand. Requires a bed6 file</li>
<li><strong>force_upper</strong>: Force uppercase output of sequences</li>
<li><strong>ignore_targets</strong>: if True, don't return any target variables</li>
</ul>
<p><strong>Output schema</strong></p>
<ul>
<li><strong>inputs</strong>: <ul>
<li><strong>shape=()</strong>, DNA sequence as string</li>
</ul>
</li>
<li><strong>targets</strong>: <ul>
<li><strong>shape=(None,)</strong>, (optional) values following the bed-entries</li>
</ul>
</li>
<li><strong>metadata</strong>: <ul>
<li><strong>ranges</strong>: <ul>
<li>Genomic ranges: chr, start, end, name, strand</li>
</ul>
</li>
</ul>
</li>
</ul>
<h2 id="seqintervaldl">SeqIntervalDl</h2>
<pre><code class="language-python">SeqIntervalDl(self, intervals_file, fasta_file, num_chr_fasta=False, label_dtype=None, auto_resize_len=None, use_strand=False, alphabet_axis=1, dummy_axis=None, alphabet='ACGT', ignore_targets=False, dtype=None)
</code></pre>
<p>Dataloader for a combination of fasta and tab-delimited input files such as bed files. The dataloader extracts regions from the fasta file as defined in the tab-delimited <code>intervals_file</code> and converts them into one-hot encoded format. Returned sequences are of the type np.array with the shape inferred from the arguments: <code>alphabet_axis</code> and <code>dummy_axis</code>.</p>
<p><strong>Arguments</strong></p>
<ul>
<li><strong>intervals_file</strong>: bed3+<columns> file path containing intervals + (optionally) labels <a href="https://raw.githubusercontent.com/kipoi/kipoiseq/master/tests/data/intervals_51bp.tsv">example</a></li>
<li><strong>fasta_file</strong>: Reference genome FASTA file path. <a href="https://raw.githubusercontent.com/kipoi/kipoiseq/master/tests/data/hg38_chr22_32000000_32300000.fa">example</a></li>
<li><strong>num_chr_fasta</strong>: True, the the dataloader will make sure that the chromosomes don't start with chr.</li>
<li><strong>label_dtype</strong>: None, datatype of the task labels taken from the intervals_file. Example: str, int, float, np.float32</li>
<li><strong>auto_resize_len</strong>: None, required sequence length.</li>
<li><strong>use_strand</strong>: reverse-complement fasta sequence if bed file defines negative strand. Requires a bed6 file</li>
<li><strong>alphabet_axis</strong>: axis along which the alphabet runs (e.g. A,C,G,T for DNA)</li>
<li><strong>dummy_axis</strong>: defines in which dimension a dummy axis should be added. None if no dummy axis is required.</li>
<li>
<p><strong>alphabet</strong>: alphabet to use for the one-hot encoding. This defines the order of the one-hot encoding. Can either be a list or a string: 'ACGT' or ['A, 'C', 'G', 'T']. Default: 'ACGT'</p>
</li>
<li>
<p><strong>dtype</strong>: defines the numpy dtype of the returned array. Example: int, np.int32, np.float32, float</p>
</li>
<li><strong>ignore_targets</strong>: if True, don't return any target variables</li>
</ul>
<p><strong>Output schema</strong></p>
<ul>
<li><strong>inputs</strong>: <ul>
<li><strong>shape=(None, 4)</strong>, One-hot encoded DNA sequence</li>
</ul>
</li>
<li><strong>targets</strong>: <ul>
<li><strong>shape=(None,)</strong>, (optional) values following the bed-entry - chr  start  end  target1   target2 ....</li>
</ul>
</li>
<li><strong>metadata</strong>: <ul>
<li><strong>ranges</strong>: <ul>
<li>Genomic ranges: chr, start, end, name, strand</li>
</ul>
</li>
</ul>
</li>
</ul>
<h2 id="anchoredgtfdl">AnchoredGTFDl</h2>
<pre><code class="language-python">AnchoredGTFDl(self, gtf_file, fasta_file, num_upstream, num_downstream, gtf_filter='gene_type == &quot;protein_coding&quot;', anchor='tss', transform=&lt;function one_hot_dna at 0x7fd04f7b8158&gt;, interval_attrs=['gene_id', 'Strand'], use_strand=True)
</code></pre>
<p>Dataloader for a combination of fasta and gtf files. The dataloader extracts fixed length regions around anchor points. Anchor points are extracted from the gtf based on the anchor parameter. The sequences corresponding to the region are then extracted from the fasta file and optionally trnasformed using a function given by the transform parameter.</p>
<p><strong>Arguments</strong></p>
<ul>
<li><strong>gtf_file</strong>: Path to a gtf file (str) <a href="https://zenodo.org/record/1466102/files/example_files-gencode.v24.annotation_chr22.gtf">example</a></li>
<li><strong>fasta_file</strong>: Reference genome FASTA file path (str) <a href="https://zenodo.org/record/1466102/files/example_files-hg38_chr22.fa">example</a></li>
<li><strong>num_upstream</strong>: Number of nt by which interval is extended upstream of the anchor point</li>
<li><strong>num_downstream</strong>: Number of nt by which interval is extended downstream of the anchor point</li>
<li>
<p><strong>gtf_filter</strong>: Allows to filter the gtf before extracting the anchor points. Can be str, callable or None. If str, it is interpreted as argument to pandas .query(). If callable, it is interpreted as function that filters a pandas dataframe and returns the filtered df.</p>
</li>
<li>
<p><strong>anchor</strong>: Defines the anchor points. Can be str or callable. If it is a callable, it is treated as function that takes a pandas dataframe and returns a modified version of the dataframe where each row represents one anchor point, the position of which is stored in the column called anchor_pos. If it is a string, a predefined function is loaded. Currently available are tss (anchor is the start of a gene), start_codon (anchor is the start of the start_codon), stop_codon (anchor is the position right after the stop_codon), polya (anchor is the position right after the end of a gene).</p>
</li>
<li>
<p><strong>transform</strong>: Callable (or None) to transform the extracted sequence (e.g. one-hot)</p>
</li>
<li><strong>interval_attrs</strong>: Metadata to extract from the gtf, e.g. ["gene_id", "Strand"]</li>
<li><strong>use_strand</strong>: True or False</li>
</ul>
<p><strong>Output schema</strong></p>
<ul>
<li><strong>inputs</strong>: <ul>
<li><strong>shape=(None, 4)</strong>, exon sequence with flanking intronic sequence</li>
</ul>
</li>
<li>
<p><strong>targets</strong>: </p>
</li>
<li>
<p><strong>metadata</strong>: </p>
<ul>
<li><strong>gene_id</strong>: <ul>
<li>gene_id</li>
</ul>
</li>
<li><strong>Strand</strong>: <ul>
<li>Strand</li>
</ul>
</li>
<li><strong>ranges</strong>: <ul>
<li>Genomic ranges: chr, start, end, name, strand</li>
</ul>
</li>
</ul>
</li>
</ul>
<h1 id="mmsplicedl">MMSpliceDl</h1>
<pre><code class="language-python">MMSpliceDl(self, gtf_file, fasta_file, intron5prime_len=100, intron3prime_len=100, transform=None, **kwargs)
</code></pre>
<p>Dataloader for splicing models. With inputs as gtf annotation file and fasta file, each output is an exon sequence with flanking intronic seuqences. Intronic sequnce lengths specified by the users. Returned sequences are of the type np.array([str])</p>
<p><strong>Arguments</strong></p>
<ul>
<li><strong>gtf_file</strong>: file path; Genome annotation GTF file <a href="https://raw.githubusercontent.com/kipoi/models/master/MMSplice/tests/data/test.gtf">example</a></li>
<li><strong>fasta_file</strong>: Reference Genome sequence in fasta format <a href="https://raw.githubusercontent.com/kipoi/models/master/MMSplice/tests/data/hg19.nochr.chr17.fa">example</a></li>
<li><strong>intron5prime_len</strong>: 5' intronic sequence length to take.</li>
<li><strong>intron3prime_len</strong>: 3' intronic sequence length to take.</li>
<li><strong>transform</strong>: transformation operation applied to the returned sequence. It needs to take seq, intron5prime_len and intron3prime_len as arguments.</li>
</ul>
<p><strong>Output schema</strong></p>
<ul>
<li><strong>inputs</strong>: <ul>
<li><strong>shape=()</strong>, exon sequence with flanking intronic sequence</li>
</ul>
</li>
<li>
<p><strong>targets</strong>: </p>
</li>
<li>
<p><strong>metadata</strong>: </p>
<ul>
<li><strong>geneID</strong>: <ul>
<li>geneID</li>
</ul>
</li>
<li><strong>transcriptID</strong>: <ul>
<li>transcriptID</li>
</ul>
</li>
<li><strong>ranges</strong>: <ul>
<li>Genomic ranges: chr, start, end, name, strand</li>
</ul>
</li>
</ul>
</li>
</ul>
              
            </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../transforms/functional/" class="btn btn-neutral float-right" title="Functional">Next <span class="icon icon-circle-arrow-right"></span></a>
      
      
        <a href=".." class="btn btn-neutral" title="Home"><span class="icon icon-circle-arrow-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
    
  </div>

  Built with <a href="https://www.mkdocs.org/">MkDocs</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
      
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" aria-label="versions">
  <span class="rst-current-version" data-toggle="rst-current-version">
    
        <span>
          <a href="http://github.com/kipoi/kipoiseq/" class="fa fa-github" style="color: #fcfcfc"> GitHub</a>
        </span>
    
    
      <span><a href=".." style="color: #fcfcfc">&laquo; Previous</a></span>
    
    
      <span><a href="../transforms/functional/" style="color: #fcfcfc">Next &raquo;</a></span>
    
  </span>
</div>
    <script>var base_url = '..';</script>
    <script src="../js/theme_extra.js" defer></script>
    <script src="../js/theme.js" defer></script>
      <script src="../search/main.js" defer></script>
    <script defer>
        window.onload = function () {
            SphinxRtdTheme.Navigation.enable(true);
        };
    </script>

</body>
</html>
