
<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
    <meta http-equiv="x-ua-compatible" content="ie=edge">
    <meta name="author" content="Nejc Zupan">
    <link rel="icon" type="image/png" href="/static/img/favicon.png">
    <title>Kipoi</title>
    <!-- Font Awesome -->
<link rel="stylesheet" type="text/css" href="/static/css/font-awesome.min.css">
<!-- Bootstrap core CSS -->
<link rel="stylesheet" type="text/css" href="/static/css/bootstrap.min.css">
<!-- Material Design Bootstrap -->
<link rel="stylesheet" type="text/css" href="/static/css/mdb.min.css">
<!-- Material Design Lite -->
<link rel="stylesheet" type="text/css" href="/static/css/material.min.css">
<!-- DataTables Material Design -->    
<link rel="stylesheet" type="text/css" href="/static/css/dataTables.material.css"/>
<!-- Highlight -->
<link rel="stylesheet" type="text/css" href="/static/css/custom_highlight_github.css"/>

<!-- Custom styles -->
<link rel="stylesheet" type="text/css" href="/static/css/style.css?v=150318">
</head>

<body>
    <section class="mb-3">
        
<!--Navbar-->

<nav class="navbar fixed-top navbar-expand-sm navbar-dark unique-color-kipoi">

    <!-- Navbar brand -->
    <span>
      <a class="nav-title-link navbar-brand-nopad text-hide left-margin" href="/"
	 ><img src="/static/img/nav_bar_logo_v3.png" height="50px" alt="Kipoi"/></a>
    </span>

    <!-- Collapse button -->
    <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarNav1" aria-controls="navbarNav1" aria-expanded="false" aria-label="Toggle navigation"><span class="navbar-toggler-icon"></span></button>
    
    <!-- Collapsible content -->
    <div class="collapse navbar-collapse text-center" id="navbarNav1">
        
        <!-- Links -->
        <ul class="navbar-nav mr-auto text-center">
	  <!-- Group list --> 
            <li class="nav-item">
              <a id="components-related-1" class="nav-button btn btn-outline-primary navbar-dark nabar-button " href="/groups/">Models</a>
            </li>
            <!-- About --> 
            <li class="nav-item">
              <a id="components-related-2" class="nav-button btn btn-outline-primary navbar-dark nabar-button " href="/about/">About</a>
            </li>
	    <!-- Docs --> 
            <li class="nav-item">
              <a id="components-related-3" class="nav-button btn btn-outline-primary navbar-dark nabar-button" href="/docs/">Docs</a>
            </li>
        <!-- Seminar -->
            <li class="nav-item">
              <a id="components-related-4" class="nav-button btn btn-outline-primary navbar-dark nabar-button " href="/seminar/">Seminar</a>
            </li>
        </ul>

        <!-- Search form -->
        

      
        <ul class="navbar-nav ml-auto text-center d-sm-none d-md-flex">
      
          <li class="nav-item">
            <a id="components-related-5" class="text-nowrap nav-link waves-effect waves-light" href="https://www.biorxiv.org/content/early/2018/07/24/375345.full.pdf">White paper<i class="fa fa-file-text-o" aria-hidden="true"></i></a>
          </li>
          <li class="nav-item">
            <a id="components-related-6" class="text-nowrap nav-link waves-effect waves-light" href="https://github.com/kipoi/models">Model repository<i class="fa fa-github" aria-hidden="true"></i></a>
          </li>
          <li class="nav-item">
            <a id="components-related-7" class="text-nowrap nav-link waves-effect waves-light" href="https://github.com/kipoi/kipoi">API repository<i class="fa fa-github" aria-hidden="true"></i></a>
          </li>
        </ul>
        
    </div>
</nav>
    </section>

    <section class="main-content">
        
    <div class="col-lg-12 col-md-12 col-sm-12">
        <!--Section: model-details content-->
        <section class="model-details">
            <!--<h2 class="">Model Details</h2>-->

                <div class="jumbotron">
                    <div class="row">
                        <div class="col-lg-7">
                            <!-- Model title -->
                            <h1 class="h1-responsive"><a href="/models/rbp_eclip">rbp_eclip</a>/SND1</h1>
                        </div>
                        <!-- Model details -->
                        <div class="col-lg-5">
                            <div class="row">
                                <div class="col-lg-6 col-md-6 col-sm-6 col-xs-12">
                                    <p>Authors: 
  
    
      
  Žiga Avsec
  
  

    
    , 
  
    
      
  Mohammadamin Barekatain
  
  

    
    , 
  
    
      
  Jun Cheng
  
  

    
    , 
  
    
      
  Julien Gagneur
  
  

    
    
  
</p>
                                    <!-- <p>
                                         Name:
                                         
					 </p> -->
                                    <!-- <p>Version: 0.1</p> -->
                                    <!-- License -->
                                    <p>License: MIT</p>
                                    <p>Contributed by: 
  
    
      
  Ziga Avsec
  
    <a href="https://github.com/avsecz" class="github">
      <i class="fa fa-github" aria-hidden="true"></i>
    </a>
  
  

    
    
  
</p>
                                    <!-- Cite as -->
                                    <p>Cite as: 
  
  
      
  
    <a href="https://doi.org/10.1093/bioinformatics/btx727">https://doi.org/10.1093/bioinformatics/btx727</a>
  

    
  
</p>
                                </div>
                                <div class="col-lg-6 col-md-6 col-xs-12">
                                    <!-- Model type -->
                                    <p>Type:
                                        
                                      <a data-toggle="tooltip" data-placement="top" data-html="true" title="
  <div>
    
      <strong>weights</strong>: RemoteFile(url=&#39;https://zenodo.org/record/1466102/files/SND1.h5?download=1&#39;, md5=&#39;1cd9249a0b65271193330983f1bfc9c9&#39;, name=&#39;&#39;)<br>
    
      <strong>custom_objects</strong>: custom_keras_objects.py<br>
    
      <strong>backend</strong>: tensorflow<br>
    
      <strong>image_dim_ordering</strong>: tf<br>
    
  </div>
"><span class="badge badge-pill rgba-stylish-strong hoverable" >keras</span></a>
                                      
                                    </p>

                                    <!-- Model postprocessing -->
                                    <p>
                                      Postprocessing:
                                      
                                      None
                                      
                                    </p>
				    
                                    <!-- Trained on -->
                                    <p>Trained on: 
  
    RBP occupancy peaks measured by eCLIP-seq (Van Nostrand et al., 2016 - https://doi.org/10.1038/nmeth.3810), https://github.com/gagneurlab/Manuscript_Avsec_Bioinformatics_2017 4000 training samples, 1000 validation sites, 1000 testing samples.

  
</p>

				    <!-- Model url  -->
				    <p>
				      <a href="https://github.com/kipoi/models/tree/master/rbp_eclip">Source files <i class="fa fa-github" aria-hidden="true"></i></a>
				    </p> 
                                </div>
                            </div>
                        </div>
                    </div>
                    <div class="row">
                        <div class="col-lg-7 col-md-12 col-sm-12">
                            <!-- Model description -->
                            <p class="lead">&#39;RBP binding model from Avsec et al: &#34;Modeling positional effects of regulatory sequences with spline transformations increases prediction accuracy of deep neural networks&#34;. &#39;
</p>
							<p>
<p>112 eCLIP models from: <a href="Avsec et al, Bioinformatics 2017">https://doi.org/10.1093/bioinformatics/btx727</a></p>
<p>Associated code is located at: <a href="https://github.com/gagneurlab/Manuscript_Avsec_Bioinformatics_2017">github.com/gagneurlab/Manuscript_Avsec_Bioinformatics_2017</a></p></p>
                        </div>

                        <div class="col-lg-5 col-md-12 col-sm-12">
                            <!-- Model code snippets -->
                            <ul class="nav nav-tabs" role="tablist">
    <li class="nav-item">
        <a class="nav-link active" data-toggle="tab" href="#cli" role="tab">CLI</a>
    </li>
    <li class="nav-item">
        <a class="nav-link" data-toggle="tab" href="#python" role="tab">python</a>
    </li>
    <li class="nav-item">
        <a class="nav-link" data-toggle="tab" href="#r" role="tab">R</a>
    </li>
    <li class="nav-item">
        <a class="nav-link" data-toggle="tab" href="#docker" role="tab">docker</a>
    </li>
    <li class="nav-item">
        <a class="nav-link" data-toggle="tab" href="#singularity" role="tab">singularity</a>
    </li>
</ul>

<!-- Tab panes -->
<div class="tab-content">
    <!-- CLI code snippet tab -->
    <div class="tab-pane active" id="cli" role="tabpanel">
        <div class="border border-top-0 code-snippet-container">
            
                
  <div class="row">
    <div class="col-lg-12 col-xs-12">
      <div class="code-title">Create a new conda environment with all dependencies installed</div>
    </div>
    <div class="col-lg-12 col-xs-12">
      <div class="code-container">
	<!-- <pre class="language-markup"><code>kipoi env create rbp_eclip
source activate kipoi-rbp_eclip</code></pre> -->
	<pre><code class="bash">kipoi env create rbp_eclip
source activate kipoi-rbp_eclip</code></pre>
	<div class="code-toolbar"><a data-toggle="tooltip" data-placement="top" data-html="true" title="Copy to clipboard!" class="btn btn-sm btn-info copy-to-clipboard">Copy</a></div>
      </div>
    </div>
  </div>

            
                
  <div class="row">
    <div class="col-lg-12 col-xs-12">
      <div class="code-title">Install model dependencies into current environment</div>
    </div>
    <div class="col-lg-12 col-xs-12">
      <div class="code-container">
	<!-- <pre class="language-markup"><code>kipoi env install rbp_eclip</code></pre> -->
	<pre><code class="bash">kipoi env install rbp_eclip</code></pre>
	<div class="code-toolbar"><a data-toggle="tooltip" data-placement="top" data-html="true" title="Copy to clipboard!" class="btn btn-sm btn-info copy-to-clipboard">Copy</a></div>
      </div>
    </div>
  </div>

            
                
  <div class="row">
    <div class="col-lg-12 col-xs-12">
      <div class="code-title">Test the model</div>
    </div>
    <div class="col-lg-12 col-xs-12">
      <div class="code-container">
	<!-- <pre class="language-markup"><code>kipoi test rbp_eclip/SND1 --source=kipoi</code></pre> -->
	<pre><code class="bash">kipoi test rbp_eclip/SND1 --source=kipoi</code></pre>
	<div class="code-toolbar"><a data-toggle="tooltip" data-placement="top" data-html="true" title="Copy to clipboard!" class="btn btn-sm btn-info copy-to-clipboard">Copy</a></div>
      </div>
    </div>
  </div>

            
                
  <div class="row">
    <div class="col-lg-12 col-xs-12">
      <div class="code-title">Make a prediction</div>
    </div>
    <div class="col-lg-12 col-xs-12">
      <div class="code-container">
	<!-- <pre class="language-markup"><code>kipoi get-example rbp_eclip/SND1 -o example
kipoi predict rbp_eclip/SND1 \
  --dataloader_args=&#39;{&#34;intervals_file&#34;: &#34;example/intervals_file&#34;, &#34;fasta_file&#34;: &#34;example/fasta_file&#34;, &#34;gtf_file&#34;: &#34;example/gtf_file&#34;, &#34;target_file&#34;: &#34;example/target_file&#34;, &#34;filter_protein_coding&#34;: true}&#39; \
  -o &#39;/tmp/rbp_eclip|SND1.example_pred.tsv&#39;
# check the results
head &#39;/tmp/rbp_eclip|SND1.example_pred.tsv&#39;
</code></pre> -->
	<pre><code class="bash">kipoi get-example rbp_eclip/SND1 -o example
kipoi predict rbp_eclip/SND1 \
  --dataloader_args=&#39;{&#34;intervals_file&#34;: &#34;example/intervals_file&#34;, &#34;fasta_file&#34;: &#34;example/fasta_file&#34;, &#34;gtf_file&#34;: &#34;example/gtf_file&#34;, &#34;target_file&#34;: &#34;example/target_file&#34;, &#34;filter_protein_coding&#34;: true}&#39; \
  -o &#39;/tmp/rbp_eclip|SND1.example_pred.tsv&#39;
# check the results
head &#39;/tmp/rbp_eclip|SND1.example_pred.tsv&#39;
</code></pre>
	<div class="code-toolbar"><a data-toggle="tooltip" data-placement="top" data-html="true" title="Copy to clipboard!" class="btn btn-sm btn-info copy-to-clipboard">Copy</a></div>
      </div>
    </div>
  </div>

            
        </div>
    </div>

    <!-- python code snippet tab -->
    <div class="tab-pane" id="python" role="tabpanel">
        <div class="border border-top-0 code-snippet-container">
            
                
  <div class="row">
    <div class="col-lg-12 col-xs-12">
      <div class="code-title">Get the model</div>
    </div>
    <div class="col-lg-12 col-xs-12">
      <div class="code-container">
	<!-- <pre class="language-markup"><code>import kipoi
model = kipoi.get_model(&#39;rbp_eclip/SND1&#39;)</code></pre> -->
	<pre><code class="python">import kipoi
model = kipoi.get_model(&#39;rbp_eclip/SND1&#39;)</code></pre>
	<div class="code-toolbar"><a data-toggle="tooltip" data-placement="top" data-html="true" title="Copy to clipboard!" class="btn btn-sm btn-info copy-to-clipboard">Copy</a></div>
      </div>
    </div>
  </div>

            
                
  <div class="row">
    <div class="col-lg-12 col-xs-12">
      <div class="code-title">Make a prediction for example files</div>
    </div>
    <div class="col-lg-12 col-xs-12">
      <div class="code-container">
	<!-- <pre class="language-markup"><code>pred = model.pipeline.predict_example(batch_size=4)</code></pre> -->
	<pre><code class="python">pred = model.pipeline.predict_example(batch_size=4)</code></pre>
	<div class="code-toolbar"><a data-toggle="tooltip" data-placement="top" data-html="true" title="Copy to clipboard!" class="btn btn-sm btn-info copy-to-clipboard">Copy</a></div>
      </div>
    </div>
  </div>

            
                
  <div class="row">
    <div class="col-lg-12 col-xs-12">
      <div class="code-title">Use dataloader and model separately</div>
    </div>
    <div class="col-lg-12 col-xs-12">
      <div class="code-container">
	<!-- <pre class="language-markup"><code># Download example dataloader kwargs
dl_kwargs = model.default_dataloader.download_example(&#39;example&#39;)
# Get the dataloader and instantiate it
dl = model.default_dataloader(**dl_kwargs)
# get a batch iterator
it = dl.batch_iter(batch_size=4)
# predict for a batch
batch = next(it)
model.predict_on_batch(batch[&#39;inputs&#39;])</code></pre> -->
	<pre><code class="python"># Download example dataloader kwargs
dl_kwargs = model.default_dataloader.download_example(&#39;example&#39;)
# Get the dataloader and instantiate it
dl = model.default_dataloader(**dl_kwargs)
# get a batch iterator
it = dl.batch_iter(batch_size=4)
# predict for a batch
batch = next(it)
model.predict_on_batch(batch[&#39;inputs&#39;])</code></pre>
	<div class="code-toolbar"><a data-toggle="tooltip" data-placement="top" data-html="true" title="Copy to clipboard!" class="btn btn-sm btn-info copy-to-clipboard">Copy</a></div>
      </div>
    </div>
  </div>

            
                
  <div class="row">
    <div class="col-lg-12 col-xs-12">
      <div class="code-title">Make predictions for custom files directly</div>
    </div>
    <div class="col-lg-12 col-xs-12">
      <div class="code-container">
	<!-- <pre class="language-markup"><code>pred = model.pipeline.predict(dl_kwargs, batch_size=4)</code></pre> -->
	<pre><code class="python">pred = model.pipeline.predict(dl_kwargs, batch_size=4)</code></pre>
	<div class="code-toolbar"><a data-toggle="tooltip" data-placement="top" data-html="true" title="Copy to clipboard!" class="btn btn-sm btn-info copy-to-clipboard">Copy</a></div>
      </div>
    </div>
  </div>

            
        </div>
    </div>
    
    <!-- R code snippet tab -->
    <div class="tab-pane" id="r" role="tabpanel">
        <div class="border border-top-0 code-snippet-container">
            
                
  <div class="row">
    <div class="col-lg-12 col-xs-12">
      <div class="code-title">Get the model</div>
    </div>
    <div class="col-lg-12 col-xs-12">
      <div class="code-container">
	<!-- <pre class="language-markup"><code>library(reticulate)
kipoi &lt;- import(&#39;kipoi&#39;)
model &lt;- kipoi$get_model(&#39;rbp_eclip/SND1&#39;)</code></pre> -->
	<pre><code class="R">library(reticulate)
kipoi &lt;- import(&#39;kipoi&#39;)
model &lt;- kipoi$get_model(&#39;rbp_eclip/SND1&#39;)</code></pre>
	<div class="code-toolbar"><a data-toggle="tooltip" data-placement="top" data-html="true" title="Copy to clipboard!" class="btn btn-sm btn-info copy-to-clipboard">Copy</a></div>
      </div>
    </div>
  </div>

            
                
  <div class="row">
    <div class="col-lg-12 col-xs-12">
      <div class="code-title">Make a prediction for example files</div>
    </div>
    <div class="col-lg-12 col-xs-12">
      <div class="code-container">
	<!-- <pre class="language-markup"><code>predictions &lt;- model$pipeline$predict_example()</code></pre> -->
	<pre><code class="R">predictions &lt;- model$pipeline$predict_example()</code></pre>
	<div class="code-toolbar"><a data-toggle="tooltip" data-placement="top" data-html="true" title="Copy to clipboard!" class="btn btn-sm btn-info copy-to-clipboard">Copy</a></div>
      </div>
    </div>
  </div>

            
                
  <div class="row">
    <div class="col-lg-12 col-xs-12">
      <div class="code-title">Use dataloader and model separately</div>
    </div>
    <div class="col-lg-12 col-xs-12">
      <div class="code-container">
	<!-- <pre class="language-markup"><code># Download example dataloader kwargs
dl_kwargs &lt;- model$default_dataloader$download_example(&#39;example&#39;)
# Get the dataloader
dl &lt;- model$default_dataloader(dl_kwargs)
# get a batch iterator
it &lt;- dl$batch_iter(batch_size=4)
# predict for a batch
batch &lt;- iter_next(it)
model$predict_on_batch(batch$inputs)</code></pre> -->
	<pre><code class="R"># Download example dataloader kwargs
dl_kwargs &lt;- model$default_dataloader$download_example(&#39;example&#39;)
# Get the dataloader
dl &lt;- model$default_dataloader(dl_kwargs)
# get a batch iterator
it &lt;- dl$batch_iter(batch_size=4)
# predict for a batch
batch &lt;- iter_next(it)
model$predict_on_batch(batch$inputs)</code></pre>
	<div class="code-toolbar"><a data-toggle="tooltip" data-placement="top" data-html="true" title="Copy to clipboard!" class="btn btn-sm btn-info copy-to-clipboard">Copy</a></div>
      </div>
    </div>
  </div>

            
                
  <div class="row">
    <div class="col-lg-12 col-xs-12">
      <div class="code-title">Make predictions for custom files directly</div>
    </div>
    <div class="col-lg-12 col-xs-12">
      <div class="code-container">
	<!-- <pre class="language-markup"><code>pred &lt;- model$pipeline$predict(dl_kwargs, batch_size=4)</code></pre> -->
	<pre><code class="R">pred &lt;- model$pipeline$predict(dl_kwargs, batch_size=4)</code></pre>
	<div class="code-toolbar"><a data-toggle="tooltip" data-placement="top" data-html="true" title="Copy to clipboard!" class="btn btn-sm btn-info copy-to-clipboard">Copy</a></div>
      </div>
    </div>
  </div>

            
        </div>
    </div>

    <!-- docker code snippet tab -->
    <div class="tab-pane" id="docker" role="tabpanel">
        <div class="border border-top-0 code-snippet-container">
            
                
  <div class="row">
    <div class="col-lg-12 col-xs-12">
      <div class="code-title">Get the docker image</div>
    </div>
    <div class="col-lg-12 col-xs-12">
      <div class="code-container">
	<!-- <pre class="language-markup"><code>docker pull kipoi/kipoi-docker:sharedpy3keras2</code></pre> -->
	<pre><code class="docker">docker pull kipoi/kipoi-docker:sharedpy3keras2</code></pre>
	<div class="code-toolbar"><a data-toggle="tooltip" data-placement="top" data-html="true" title="Copy to clipboard!" class="btn btn-sm btn-info copy-to-clipboard">Copy</a></div>
      </div>
    </div>
  </div>

            
                
  <div class="row">
    <div class="col-lg-12 col-xs-12">
      <div class="code-title">Get the activated conda environment inside the container</div>
    </div>
    <div class="col-lg-12 col-xs-12">
      <div class="code-container">
	<!-- <pre class="language-markup"><code>docker run -it kipoi/kipoi-docker:sharedpy3keras2</code></pre> -->
	<pre><code class="docker">docker run -it kipoi/kipoi-docker:sharedpy3keras2</code></pre>
	<div class="code-toolbar"><a data-toggle="tooltip" data-placement="top" data-html="true" title="Copy to clipboard!" class="btn btn-sm btn-info copy-to-clipboard">Copy</a></div>
      </div>
    </div>
  </div>

            
                
  <div class="row">
    <div class="col-lg-12 col-xs-12">
      <div class="code-title">Test the model</div>
    </div>
    <div class="col-lg-12 col-xs-12">
      <div class="code-container">
	<!-- <pre class="language-markup"><code>docker run kipoi/kipoi-docker:sharedpy3keras2 kipoi test rbp_eclip/SND1 --source=kipoi</code></pre> -->
	<pre><code class="docker">docker run kipoi/kipoi-docker:sharedpy3keras2 kipoi test rbp_eclip/SND1 --source=kipoi</code></pre>
	<div class="code-toolbar"><a data-toggle="tooltip" data-placement="top" data-html="true" title="Copy to clipboard!" class="btn btn-sm btn-info copy-to-clipboard">Copy</a></div>
      </div>
    </div>
  </div>

            
                
  <div class="row">
    <div class="col-lg-12 col-xs-12">
      <div class="code-title">Make prediction for custom files directly</div>
    </div>
    <div class="col-lg-12 col-xs-12">
      <div class="code-container">
	<!-- <pre class="language-markup"><code># Create an example directory containing the data
mkdir -p $PWD/kipoi-example 
# You can replace $PWD/kipoi-example with a different absolute path containing the data 
docker run -v $PWD/kipoi-example:/app/ kipoi/kipoi-docker:sharedpy3keras2 \
kipoi get-example rbp_eclip/SND1 -o /app/example 
docker run -v $PWD/kipoi-example:/app/ kipoi/kipoi-docker:sharedpy3keras2 \
kipoi predict rbp_eclip/SND1 \
--dataloader_args=&#39;{&#39;intervals_file&#39;: &#39;/app/example/intervals_file&#39;, &#39;fasta_file&#39;: &#39;/app/example/fasta_file&#39;, &#39;gtf_file&#39;: &#39;/app/example/gtf_file&#39;, &#39;target_file&#39;: &#39;/app/example/target_file&#39;, &#39;filter_protein_coding&#39;: True}&#39; \
-o &#39;/app/rbp_eclip_SND1.example_pred.tsv&#39; 
# check the results
head $PWD/kipoi-example/rbp_eclip_SND1.example_pred.tsv
</code></pre> -->
	<pre><code class="docker"># Create an example directory containing the data
mkdir -p $PWD/kipoi-example 
# You can replace $PWD/kipoi-example with a different absolute path containing the data 
docker run -v $PWD/kipoi-example:/app/ kipoi/kipoi-docker:sharedpy3keras2 \
kipoi get-example rbp_eclip/SND1 -o /app/example 
docker run -v $PWD/kipoi-example:/app/ kipoi/kipoi-docker:sharedpy3keras2 \
kipoi predict rbp_eclip/SND1 \
--dataloader_args=&#39;{&#39;intervals_file&#39;: &#39;/app/example/intervals_file&#39;, &#39;fasta_file&#39;: &#39;/app/example/fasta_file&#39;, &#39;gtf_file&#39;: &#39;/app/example/gtf_file&#39;, &#39;target_file&#39;: &#39;/app/example/target_file&#39;, &#39;filter_protein_coding&#39;: True}&#39; \
-o &#39;/app/rbp_eclip_SND1.example_pred.tsv&#39; 
# check the results
head $PWD/kipoi-example/rbp_eclip_SND1.example_pred.tsv
</code></pre>
	<div class="code-toolbar"><a data-toggle="tooltip" data-placement="top" data-html="true" title="Copy to clipboard!" class="btn btn-sm btn-info copy-to-clipboard">Copy</a></div>
      </div>
    </div>
  </div>

            
        </div>
    </div>

    <!-- singularity code snippet tab -->
    <div class="tab-pane" id="singularity" role="tabpanel">
        <div class="border border-top-0 code-snippet-container">
            
                
  <div class="row">
    <div class="col-lg-12 col-xs-12">
      <div class="code-title">Get the singularity image</div>
    </div>
    <div class="col-lg-12 col-xs-12">
      <div class="code-container">
	<!-- <pre class="language-markup"><code>https://zenodo.org/record/5644007/files/kipoi-docker_sharedpy3keras2.sif?download=1</code></pre> -->
	<pre><code class="singularity">https://zenodo.org/record/5644007/files/kipoi-docker_sharedpy3keras2.sif?download=1</code></pre>
	<div class="code-toolbar"><a data-toggle="tooltip" data-placement="top" data-html="true" title="Copy to clipboard!" class="btn btn-sm btn-info copy-to-clipboard">Copy</a></div>
      </div>
    </div>
  </div>

            
                
  <div class="row">
    <div class="col-lg-12 col-xs-12">
      <div class="code-title">Test the model</div>
    </div>
    <div class="col-lg-12 col-xs-12">
      <div class="code-container">
	<!-- <pre class="language-markup"><code>singularity exec kipoi-docker_sharedpy3keras2 kipoi test rbp_eclip/SND1 --source=kipoi</code></pre> -->
	<pre><code class="singularity">singularity exec kipoi-docker_sharedpy3keras2 kipoi test rbp_eclip/SND1 --source=kipoi</code></pre>
	<div class="code-toolbar"><a data-toggle="tooltip" data-placement="top" data-html="true" title="Copy to clipboard!" class="btn btn-sm btn-info copy-to-clipboard">Copy</a></div>
      </div>
    </div>
  </div>

            
                
  <div class="row">
    <div class="col-lg-12 col-xs-12">
      <div class="code-title">Make prediction for custom files directly</div>
    </div>
    <div class="col-lg-12 col-xs-12">
      <div class="code-container">
	<!-- <pre class="language-markup"><code># Create an example directory containing the data
mkdir -p $PWD/kipoi-example 
# You can replace $PWD/kipoi-example with a different absolute path containing the data 
singularity exec kipoi-docker_sharedpy3keras2 \
kipoi get-example rbp_eclip/SND1 -o $PWD/kipoi-example/ 
singularity exec kipoi-docker_sharedpy3keras2 \
kipoi predict rbp_eclip/SND1 \
--dataloader_args=&#39;{&#39;intervals_file&#39;: &#39;example/intervals_file&#39;, &#39;fasta_file&#39;: &#39;example/fasta_file&#39;, &#39;gtf_file&#39;: &#39;example/gtf_file&#39;, &#39;target_file&#39;: &#39;example/target_file&#39;, &#39;filter_protein_coding&#39;: True}&#39; \
-o &#39;rbp_eclip_SND1.example_pred.tsv&#39; 
# check the results
head $PWD/kipoi-example/rbp_eclip_SND1.example_pred.tsv
</code></pre> -->
	<pre><code class="singularity"># Create an example directory containing the data
mkdir -p $PWD/kipoi-example 
# You can replace $PWD/kipoi-example with a different absolute path containing the data 
singularity exec kipoi-docker_sharedpy3keras2 \
kipoi get-example rbp_eclip/SND1 -o $PWD/kipoi-example/ 
singularity exec kipoi-docker_sharedpy3keras2 \
kipoi predict rbp_eclip/SND1 \
--dataloader_args=&#39;{&#39;intervals_file&#39;: &#39;example/intervals_file&#39;, &#39;fasta_file&#39;: &#39;example/fasta_file&#39;, &#39;gtf_file&#39;: &#39;example/gtf_file&#39;, &#39;target_file&#39;: &#39;example/target_file&#39;, &#39;filter_protein_coding&#39;: True}&#39; \
-o &#39;rbp_eclip_SND1.example_pred.tsv&#39; 
# check the results
head $PWD/kipoi-example/rbp_eclip_SND1.example_pred.tsv
</code></pre>
	<div class="code-toolbar"><a data-toggle="tooltip" data-placement="top" data-html="true" title="Copy to clipboard!" class="btn btn-sm btn-info copy-to-clipboard">Copy</a></div>
      </div>
    </div>
  </div>

            
        </div>
    </div>

</div>
                        </div>
                    </div>
                    
                    <hr class="w-100 clearfix">

                    <div class="row">
                        <div class="col-lg-5 col-md-12">
                            <!-- Model schema -->
                            <div class="card">
    <div class="card-header primary-color white-text">
        Schema
    </div>
    <div class="card-body">
        <div class="row">
            <div class="col-lg-6 col-md-6 col-sm-12 col-xs-12">
                <h4 class="card-title mt-2 mb-3">Inputs</h4>
                
  <p>Dictionary of numpy arrays</p>
  
    <p><strong>Name</strong>: seq</p>
    <p>
      &nbsp;&nbsp;&nbsp;&nbsp;<strong>Shape</strong>: (101, 4)&nbsp;<a data-toggle="tooltip" data-placement="top" data-html="true" title="Numpy array shape for a single data point (i.e. batch dimension excluded)."><i class="fa fa-question-circle blue-grey-text" style="font-size: 0.9em;" aria-hidden="true"></i></a></p>
    <p>&nbsp;&nbsp;&nbsp;&nbsp;<strong>Doc</strong>: One-hot encoded RNA sequence</p>

    <p><strong>Name</strong>: dist_tss</p>
    <p>
      &nbsp;&nbsp;&nbsp;&nbsp;<strong>Shape</strong>: (1, 10)&nbsp;<a data-toggle="tooltip" data-placement="top" data-html="true" title="Numpy array shape for a single data point (i.e. batch dimension excluded)."><i class="fa fa-question-circle blue-grey-text" style="font-size: 0.9em;" aria-hidden="true"></i></a></p>
    <p>&nbsp;&nbsp;&nbsp;&nbsp;<strong>Doc</strong>: Distance the nearest TSS site transformed with B-splines</p>

    <p><strong>Name</strong>: dist_polya</p>
    <p>
      &nbsp;&nbsp;&nbsp;&nbsp;<strong>Shape</strong>: (1, 10)&nbsp;<a data-toggle="tooltip" data-placement="top" data-html="true" title="Numpy array shape for a single data point (i.e. batch dimension excluded)."><i class="fa fa-question-circle blue-grey-text" style="font-size: 0.9em;" aria-hidden="true"></i></a></p>
    <p>&nbsp;&nbsp;&nbsp;&nbsp;<strong>Doc</strong>: Distance the nearest Poly-A site transformed with B-splines</p>

    <p><strong>Name</strong>: dist_exon_intron</p>
    <p>
      &nbsp;&nbsp;&nbsp;&nbsp;<strong>Shape</strong>: (1, 10)&nbsp;<a data-toggle="tooltip" data-placement="top" data-html="true" title="Numpy array shape for a single data point (i.e. batch dimension excluded)."><i class="fa fa-question-circle blue-grey-text" style="font-size: 0.9em;" aria-hidden="true"></i></a></p>
    <p>&nbsp;&nbsp;&nbsp;&nbsp;<strong>Doc</strong>: Distance the nearest exon_intron (splice donor) site transformed with B-splines</p>

    <p><strong>Name</strong>: dist_intron_exon</p>
    <p>
      &nbsp;&nbsp;&nbsp;&nbsp;<strong>Shape</strong>: (1, 10)&nbsp;<a data-toggle="tooltip" data-placement="top" data-html="true" title="Numpy array shape for a single data point (i.e. batch dimension excluded)."><i class="fa fa-question-circle blue-grey-text" style="font-size: 0.9em;" aria-hidden="true"></i></a></p>
    <p>&nbsp;&nbsp;&nbsp;&nbsp;<strong>Doc</strong>: Distance the nearest intron_exon (splice acceptor) site transformed with B-splines</p>

    <p><strong>Name</strong>: dist_start_codon</p>
    <p>
      &nbsp;&nbsp;&nbsp;&nbsp;<strong>Shape</strong>: (1, 10)&nbsp;<a data-toggle="tooltip" data-placement="top" data-html="true" title="Numpy array shape for a single data point (i.e. batch dimension excluded)."><i class="fa fa-question-circle blue-grey-text" style="font-size: 0.9em;" aria-hidden="true"></i></a></p>
    <p>&nbsp;&nbsp;&nbsp;&nbsp;<strong>Doc</strong>: Distance the nearest start codon transformed with B-splines</p>

    <p><strong>Name</strong>: dist_stop_codon</p>
    <p>
      &nbsp;&nbsp;&nbsp;&nbsp;<strong>Shape</strong>: (1, 10)&nbsp;<a data-toggle="tooltip" data-placement="top" data-html="true" title="Numpy array shape for a single data point (i.e. batch dimension excluded)."><i class="fa fa-question-circle blue-grey-text" style="font-size: 0.9em;" aria-hidden="true"></i></a></p>
    <p>&nbsp;&nbsp;&nbsp;&nbsp;<strong>Doc</strong>: Distance the nearest stop codon transformed with B-splines</p>

    <p><strong>Name</strong>: dist_gene_start</p>
    <p>
      &nbsp;&nbsp;&nbsp;&nbsp;<strong>Shape</strong>: (1, 10)&nbsp;<a data-toggle="tooltip" data-placement="top" data-html="true" title="Numpy array shape for a single data point (i.e. batch dimension excluded)."><i class="fa fa-question-circle blue-grey-text" style="font-size: 0.9em;" aria-hidden="true"></i></a></p>
    <p>&nbsp;&nbsp;&nbsp;&nbsp;<strong>Doc</strong>: Distance the nearest gene start transformed with B-splines</p>

    <p><strong>Name</strong>: dist_gene_end</p>
    <p>
      &nbsp;&nbsp;&nbsp;&nbsp;<strong>Shape</strong>: (1, 10)&nbsp;<a data-toggle="tooltip" data-placement="top" data-html="true" title="Numpy array shape for a single data point (i.e. batch dimension excluded)."><i class="fa fa-question-circle blue-grey-text" style="font-size: 0.9em;" aria-hidden="true"></i></a></p>
    <p>&nbsp;&nbsp;&nbsp;&nbsp;<strong>Doc</strong>: Distance the nearest gene end transformed with B-splines</p>


            </div>
            <hr class="w-100 clearfix d-md-none">
            <div class="col-lg-6 col-md-6 col-sm-12 col-xs-12">
                <h4 class="card-title mt-2 mb-3">Targets</h4>
                
  <p>Single numpy array</p>
  
    <p><strong>Name</strong>: None</p>
    <p>
      &nbsp;&nbsp;&nbsp;&nbsp;<strong>Shape</strong>: (1,)&nbsp;<a data-toggle="tooltip" data-placement="top" data-html="true" title="Numpy array shape for a single data point (i.e. batch dimension excluded)."><i class="fa fa-question-circle blue-grey-text" style="font-size: 0.9em;" aria-hidden="true"></i></a></p>
    <p>&nbsp;&nbsp;&nbsp;&nbsp;<strong>Doc</strong>: Predicted binding strength</p>


            </div>
        </div>
    </div>
</div>
                        </div>
                        <hr class="w-100 clearfix d-lg-none">
                        
                            <div class="col-lg-7 col-md-12">
                                <!-- Dataloader details -->
                                <div class="card">
        <div class="card-header warning-color-dark white-text clearfix">
            Dataloader
        </div>
        <div class="card-body">
            <div class="row">
              <div class="col-lg-5">
		<!-- TODO - ifelse  -->
		
  		  <p><strong>Defined as:</strong> .</p>
		
                  <p><strong>Doc:</strong> RBP binding model taking as input 101nt long sequence as well as 8 distances to nearest genomic landmarks -  tss, poly-A, exon-intron boundary, intron-exon boundary, start codon, stop codon, gene start, gene end </p>
                  <p><strong>Authors:</strong> 
  
    
      
  Ziga Avsec
  
    <a href="https://github.com/avsecz" class="github">
      <i class="fa fa-github" aria-hidden="true"></i>
    </a>
  
  

    
    
  
</p>
                  <p><strong>Type:</strong> None</p>
                  <p><strong>License:</strong> MIT</p>

                </div>
                <hr class="w-100 clearfix d-lg-none">
                <div class="col-lg-7">
                  <h4 class="card-title mt-2 mb-3">Arguments</h4>
                  
  <div>
    
      <p>
	<strong>intervals_file</strong> :
	bed6 file with `chrom start end id score strand` columns
      </p>
    
      <p>
	<strong>fasta_file</strong> :
	Reference genome sequence
      </p>
    
      <p>
	<strong>gtf_file</strong> :
	file path; Genome annotation GTF file
      </p>
    
      <p>
	<strong>target_file</strong> (optional):
	path to the targets (txt) file
      </p>
    
      <p>
	<strong>filter_protein_coding</strong> (optional):
	Considering genomic landmarks only for protein coding genes when computing the distances to the nearest genomic landmark.
      </p>
    
      <p>
	<strong>position_transformer_file</strong> :
	Provided file for each dataloader obtained from https://sandbox.zenodo.org/record/248594/files/SND1.dataloader.pkl?download=1
      </p>
    
      <p>
	<strong>use_linecache</strong> (optional):
	if True, use linecache https://docs.python.org/3/library/linecache.html to access bed file rows
      </p>
    
  </div>

                </div>
            </div>
        </div>
</div>
    
                            </div>
                        
                    </div>
                
                    <hr class="w-100 clearfix">
                    
                    <!-- Model and dataloader dependencies -->
                    <div class="row">
                        <div class="col-lg-12 col-md-12 col-xs-12">
                            <div class="row">
                                <div class="col-lg-5 col-md-5 col-sm-12 col-xs-12">
    <div class="card">
        <div class="card-header">
            Model dependencies
        </div>
        <div class="card-body">
            <div class="row">
                <div class="col-lg-6 col-md-6 col-sm-6 col-xs-12">
                    <strong>conda</strong>:<br>
                    
  <ul class="dependencies">
    
      <li>pip=20.3.3</li>
    
</u>

                </div>
                <hr class="w-100 clearfix d-sm-none">
                <div class="col-lg-6 col-md-6 col-sm-6 col-xs-12">
                    <strong>pip</strong>:<br>
                    
  <ul class="dependencies">
    
      <li>concise==0.6.9<br></li>
    
      <li>tensorflow==1.4.1<br></li>
    
      <li>keras==2.1.6<br></li>
    
      <li>numpy==1.17.4<br></li>
    
      <li>h5py==2.10.0</li>
    
</u>

                </div>
            </div>
        </div>
    </div>
</div>
<hr class="w-100 clearfix d-md-none">
<div class="col-lg-7 col-md-7 col-sm-12 col-xs-12">
    <div class="card">
        <div class="card-header">
            Dataloader dependencies
        </div>
        <div class="card-body">
            <div class="row">
                
                    <div class="col-lg-6 col-md-6 col-sm-6 col-xs-12">
                        <strong>conda</strong>:<br>
                        
  <ul class="dependencies">
    
      <li>bioconda::pybedtools<br></li>
    
      <li>bioconda::pysam=0.15.3<br></li>
    
      <li>bioconda::genomelake=0.1.4<br></li>
    
      <li>bioconda::pybigwig<br></li>
    
      <li>bioconda::pyfaidx<br></li>
    
      <li>bioconda::gtfparse&gt;=1.0.7<br></li>
    
      <li>python=3.6<br></li>
    
      <li>cython<br></li>
    
      <li>scikit-learn=0.18.1<br></li>
    
      <li>pandas=0.23.4<br></li>
    
      <li>pip=20.3.3</li>
    
</u>

                    </div>
                    <hr class="w-100 clearfix d-sm-none">
                    <div class="col-lg-6 col-md-6 col-sm-6 col-xs-12">
                        <strong>pip</strong>:<br>
                        
  <ul class="dependencies">
    
      <li>concise&gt;=0.6.6<br></li>
    
      <li>numpy&gt;=1.13.3</li>
    
</u>

                    </div>
                
            </div>
        </div>
    </div>
</div>
                            </div>
                        </div>
                    </div>
                    <hr class="w-100 clearfix">
                    <div class="row">
                        <div class="col-lg-12 col-md-12 col-xs-12">
                          <div id="disqus_thread"></div>
                        </div>
                    </div>
                </div>
        </section>
        <!--/Section: model-details content-->                        
    </div>

    </section>
    
    <footer class="page-footer center-on-small-only unique-color">
    <!--Footer links-->
    <div class="container-fluid">
        <div class="row">
            <!--First column-->
            <div class="col-lg-3 col-md-6 ml-auto">
                <h5 class="title mb-3"><strong>About KIPOI project</strong></h5>
                <p>Kipoi is an API and a repository of ready-to-use trained models for genomics.</p>
            </div>
            <!--/.First column-->
            <hr class="w-100 clearfix d-md-none">
            <!--Second column-->
            <div class="col-lg-2 col-md-6 ml-auto">
              <h5 class="title mb-3"><strong>Useful links</strong></h5>
              <ul>
                <li>
                  <a href="https://github.com/kipoi/kipoi">API repository <i class="fa fa-github" aria-hidden="true"></i></a>
                </li>
                <li>
                  <a href="https://github.com/kipoi/models">Model repository <i class="fa fa-github" aria-hidden="true"></i></a>
                </li>
                <li>
                  <a href="https://github.com/kipoi/website">Website repository <i class="fa fa-github" aria-hidden="true"></i></a>
                </li>
              </ul>
            </div>
            <!--/.Second column-->
            <hr class="w-100 clearfix d-lg-none">
            <!--Third column-->
            <div class="col-lg-4 col-md-12 ml-auto text-center">
                <ul>
                    <li>
                        <h5>Like the project?</h5>
                    </li>
                    <li>
                        <a target="_blank" href="https://twitter.com/KipoiZoo" class="btn btn-primary waves-effect waves-light" rel="nofollow">Tweet us!</a>
                    </li>
                </ul>
            </div>
            <!--/.Third column-->
        </div>
    </div>
    <!--/.Footer links-->

    <!--Copyright-->
    <div class="footer-copyright">
        <div class="containter-fluid">
            © 2021 The Kipoi team, website developed by 
            <a href="https://github.com/zupan" class="Nejc Zupan github">
                Nejc Zupan
                <i class="fa fa-github" aria-hidden="true"></i>
            </a>
        </div>
    </div>
    <!--/.Copyright-->
</footer>

    <!-- SCRIPTS -->
<!-- JQuery -->
<script type="text/javascript" src="/static/js/jquery-3.2.1.min.js"></script>
<!-- Bootstrap tooltips -->
<script type="text/javascript" src="/static/js/popper.min.js"></script>
<!-- Bootstrap core JavaScript -->
<script type="text/javascript" src="/static/js/bootstrap.min.js"></script>
<!-- MDB core JavaScript -->
<script type="text/javascript" src="/static/js/mdb.min.js"></script>

<!-- DataTables -->
<script type="text/javascript" src="/static/js/jquery.dataTables.min.js"></script>
<script type="text/javascript" src="/static/js/dataTables.material.min.js"></script>

<!-- Highlight -->
<script type="text/javascript" src="/static/js/highlight.pack.js"></script>

<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-115025493-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-115025493-1');
</script>
    
    
    <script language="JavaScript">
      // Enable code highlighting
      hljs.initHighlightingOnLoad();

      $(document).ready(function() {
        $('[data-toggle="tooltip"]').tooltip();

        $('.copy-to-clipboard').on('click', function() {
          /* Get the code field */
          var copyCode = $(this).closest('.code-container').find('code').text();
          var $temp = $("<textarea>");
          $("body").append($temp);
          $temp.val(copyCode).select();
          document.execCommand("copy");
          $temp.remove();
        });
      });

      /**
       *  RECOMMENDED CONFIGURATION VARIABLES: EDIT AND UNCOMMENT THE SECTION BELOW TO INSERT DYNAMIC VALUES FROM YOUR PLATFORM OR CMS.
       *  LEARN WHY DEFINING THESE VARIABLES IS IMPORTANT: https://disqus.com/admin/universalcode/#configuration-variables*/
      
      var disqus_config = function () {
        this.page.url = "http://localhost/models/rbp_eclip/SND1/";  // Replace PAGE_URL with your page's canonical URL variable
        this.page.identifier = "/models/rbp_eclip/SND1/"; // Replace PAGE_IDENTIFIER with your page's unique identifier variable
      };
      
      (function() { // DON'T EDIT BELOW THIS LINE
        var d = document, s = d.createElement('script');
        s.src = 'https://kipoi-org.disqus.com/embed.js';
        s.setAttribute('data-timestamp', +new Date());
        (d.head || d.body).appendChild(s);
      })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    

</body>

</html>